{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9638fa9d",
   "metadata": {},
   "source": [
    "### 读取数据"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "08158ecf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "\n",
    "data_dir=r'D:\\项目资料\\基因表达\\通路分析'\n",
    "file_gene_path=os.path.join(data_dir, 'gene_alias.txt')\n",
    "file_wnt4_path=os.path.join(data_dir, '(cancer) AND (WNT4)文献.txt')\n",
    "file_ca_path = os.path.join(data_dir, '(cancer) AND (Carbonic Anhydrase 1)文献.txt')\n",
    "\n",
    "def read_gene_data_file(file_path):\n",
    "    data = []\n",
    "    whole_line = ''\n",
    "    with open(file_path) as f:\n",
    "        for line in f.readlines():\n",
    "            line = line[:-1]\n",
    "            whole_line = whole_line + line\n",
    "            if not line.endswith(' '):\n",
    "                data.append(whole_line)\n",
    "                whole_line = ''\n",
    "        data.append(whole_line)\n",
    "    return data\n",
    "\n",
    "def sel_item(gene_data):\n",
    "    result, item = [], []\n",
    "    start_nbr, i = 1, 0\n",
    "    while i < len(gene_data):\n",
    "        if gene_data[i].startswith('%s.' % start_nbr):\n",
    "#             print('start nbr: %s match!' % start_nbr)\n",
    "            if len(item) > 0:\n",
    "                result.append(item)\n",
    "                item = []\n",
    "            \n",
    "            start_nbr = start_nbr + 1\n",
    "        item.append(gene_data[i])\n",
    "        i = i + 1\n",
    "        \n",
    "    if len(item) > 0:\n",
    "        result.append(item)\n",
    "            \n",
    "    print(len(result))\n",
    "    return result\n",
    "\n",
    "def proc_item(item_arr):\n",
    "    results = []\n",
    "    for item_idx, item in enumerate(item_arr):\n",
    "        starts_search_flag = 'Author information:'\n",
    "        ends_search_flag = '(^DOI:)|(^©)|(^PMID: )|(^Copyright)'\n",
    "        # 标题\n",
    "        result_item = [item[2]]\n",
    "        for idx, line_str in enumerate(item):\n",
    "            # 找Author Information\n",
    "            if re.search(starts_search_flag, line_str):\n",
    "                # 找空行\n",
    "                no = idx + 1\n",
    "                while not item[no] == '':\n",
    "                    no = no + 1\n",
    "                no = no + 1\n",
    "                \n",
    "                # 摘要前可能有Comment in\n",
    "                if item[no].startswith('Comment in') or item[no].startswith('Erratum in'):\n",
    "                    while not item[no] == '':\n",
    "                        no = no + 1\n",
    "                    no = no + 1\n",
    "                \n",
    "                # 找摘要\n",
    "                while not item[no] == '':\n",
    "                    result_item.append(item[no])\n",
    "                    no = no + 1\n",
    "                    \n",
    "                results.append(result_item)\n",
    "                break\n",
    "                \n",
    "            # 找到结束记号\n",
    "            if re.search(ends_search_flag, line_str):\n",
    "                # 找空行\n",
    "                ends_no = idx - 1\n",
    "                starts_no = idx - 2\n",
    "                while not item[starts_no] == '':\n",
    "                    starts_no = starts_no - 1\n",
    "                    \n",
    "                # 写内容\n",
    "                result_item.extend(item[starts_no + 1:ends_no])\n",
    "                \n",
    "                results.append(result_item)\n",
    "                break\n",
    "                \n",
    "            \n",
    "    return results\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "208ffad6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import random\n",
    "def write_split_lines(data, train_path, val_path):\n",
    "    results = []\n",
    "    for item in data:\n",
    "        for line in item:\n",
    "            if len(line.strip()) > 10:\n",
    "                results.append(line)\n",
    "    \n",
    "    random.shuffle(results)\n",
    "    train_len = int(len(results) * 0.8)\n",
    "    train_data, val_data = results[:train_len], results[train_len:]\n",
    "    with open(train_path, 'w') as f:\n",
    "        for line in train_data:\n",
    "            f.write('%s\\n' % line)\n",
    "            \n",
    "    with open(val_path, 'w') as f:\n",
    "        for line in val_data:\n",
    "            f.write('%s\\n' % line)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f96e10cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(cancer) AND (c-myc)文献19810409-20040827.txt\n",
      "9052\n",
      "(cancer) AND (c-myc)文献20040827-20160827.txt\n",
      "9757\n",
      "(cancer) AND (c-myc)文献20160827-20220827.txt\n",
      "8089\n",
      "(cancer) AND (P53)文献19790501-19970827.txt\n",
      "8217\n",
      "(cancer) AND (P53)文献19970827-20000827.txt\n",
      "7429\n",
      "(cancer) AND (P53)文献20000827-20030827.txt\n",
      "7938\n",
      "(cancer) AND (P53)文献20030827-20060827.txt\n",
      "8142\n",
      "(cancer) AND (P53)文献20060827-20090827.txt\n",
      "8454\n",
      "(cancer) AND (P53)文献20090827-20120827.txt\n",
      "9746\n",
      "(cancer) AND (P53)文献20120827-20140827.txt\n",
      "7483\n",
      "(cancer) AND (P53)文献20140827-20160827.txt\n",
      "7515\n",
      "(cancer) AND (P53)文献20160827-20180827.txt\n",
      "7459\n",
      "(cancer) AND (P53)文献20180827-20200827.txt\n",
      "7667\n",
      "(cancer) AND (P53)文献20200827-20220827.txt\n",
      "7601\n",
      "(cancer) AND (STAT)文献19470201-20150827.txt\n",
      "7479\n",
      "(cancer) AND (STAT)文献20150827-20220827.txt\n",
      "5687\n",
      "(cancer) AND PDL1)文献20001002-20191028.txt\n",
      "6721\n",
      "(cancer) AND PDL1)文献20191008-20220827.txt\n",
      "9872\n"
     ]
    }
   ],
   "source": [
    "base_dir = r'D:\\projects\\NLP\\Gene\\Data'\n",
    "files = os.listdir(base_dir)\n",
    "files = [file for file in files if '文献' in file]\n",
    "files.remove('(cancer) AND (WNT4)文献.txt')\n",
    "files.remove('(cancer) AND (Carbonic Anhydrase 1)文献.txt')\n",
    "\n",
    "data = []\n",
    "for file in files:\n",
    "    print(file)\n",
    "    data.extend(sel_item(read_gene_data_file(os.path.join(base_dir, file))))\n",
    "write_split_lines(data, os.path.join(base_dir, 'train.txt'), os.path.join(base_dir, 'val.txt'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1707f4e9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1. Adv Exp Med Biol. 2020;1248:33-59. doi: 10.1007/978-981-15-3266-5_3.',\n",
       " '',\n",
       " 'Roles of PD-1/PD-L1 Pathway: Signaling, Cancer, and Beyond.',\n",
       " '',\n",
       " 'Ai L(1), Xu A(2), Xu J(3).',\n",
       " '',\n",
       " 'Author information:',\n",
       " '(1)Department of Medical Oncology, Zhongshan Hospital, Fudan University, Shanghai, 200032, China. storysparrow@126.com.',\n",
       " '(2)Department of Rheumatology, Renji Hospital, Shanghai Jiaotong University, Shanghai, 200001, China.',\n",
       " '(3)Institutes of Biomedical Sciences, Zhongshan-Xuhui Hospital, Fudan University, Shanghai, 200032, China. jie_xu@fudan.edu.cn.',\n",
       " '',\n",
       " 'Immunotherapies that target PD-1/PD-L1 axis have shown unprecedented success in a wide variety of human cancers. PD-1 is one of the key coinhibitory receptors expressed on T cells upon T cell activation. After engagement with its ligands, mainly PD-L1, PD-1 is activated and recruits the phosphatase SHP-2 in proximity to T cell receptor (TCR) and CD28 signaling. This event results in dephosphorylation and attenuation of key molecules in TCR and CD28 pathway, leading to inhibition of T cell proliferation, activation, cytokine production, altered metabolism and cytotoxic T lymphocytes (CTLs) killer functions, and eventual death of activated T cells. Bodies evolve coinhibitory pathways controlling T cell response magnitude and duration to limit tissue damage and maintain self-tolerance. However, tumor cells hijack these inhibitory pathways to escape host immune surveillance by overexpression of PD-L1. This provides the scientific rationale for clinical application of immune checkpoint inhibitors in oncology. The aberrantly high expression of PD-L1 in tumor microenvironment (TME) can be attributable to the \"primary\" activation of multiple oncogenic signaling and the \"secondary\" induction by inflammatory factors such as IFN-γ. Clinically, antibodies targeting PD-1/PD-L1 reinvigorate the \"exhausted\" T cells in TME and show remarkable objective response and durable remission with acceptable toxicity profile in large numbers of tumors such as melanoma, lymphoma, and mismatch-repair deficient tumors. Nevertheless, most patients are still refractory to anti-PD-1/PD-L1 therapy. Identifying the predictive biomarkers and design rational PD-1-based combination therapy become the priorities in cancer immunotherapy. PD-L1 expression, cytotoxic T lymphocytes infiltration, and tumor mutation burden (TMB) are generally considered as the most important factors affecting the effectiveness of PD-1/PD-L1 blockade. The revolution in cancer immunotherapy achieved by PD-1/PD-L1 blockade offers the paradigm for scientific translation from bench to bedside. The next decades will without doubt witness the renaissance of immunotherapy.',\n",
       " '',\n",
       " 'DOI: 10.1007/978-981-15-3266-5_3',\n",
       " 'PMID: 32185706 [Indexed for MEDLINE]',\n",
       " '',\n",
       " '']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "beaee3fc",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "245\n",
      "1818\n"
     ]
    }
   ],
   "source": [
    "# data_wnt4 = proc_item(sel_item(read_gene_data_file(file_wnt4_path)))\n",
    "# data_ca = proc_item(sel_item(read_gene_data_file(file_ca_path)))\n",
    "data_wnt4 = sel_item(read_gene_data_file(file_wnt4_path))\n",
    "data_ca = sel_item(read_gene_data_file(file_ca_path))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "a6ab5f66",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import xlwt\n",
    "# workbook = xlwt.Workbook()\n",
    "# sheet_ca = workbook.add_sheet(\"Sheet_ca\")\n",
    "# for idx, item in enumerate(data_ca):\n",
    "#     sheet_ca.write(idx, 0, '\\n'.join(item))\n",
    "# sheet_wnt4 = workbook.add_sheet(\"Sheet_wnt4\")\n",
    "# for idx, item in enumerate(data_wnt4):\n",
    "#     sheet_wnt4.write(idx, 0, '\\n'.join(item))\n",
    "# workbook.save(r'基因摘要表格.xlsx')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "774a5ef4",
   "metadata": {},
   "source": [
    "### 生成Label Studio标记数据"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d427d1b1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import random\n",
    "data_all = data_wnt4 + data_ca\n",
    "random.seed(2345)\n",
    "random.shuffle(data_all)\n",
    "train_len = int(len(data_all) * 0.8)\n",
    "train_data, test_data = data_all[:train_len], data_all[train_len:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "01f5389e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import json\n",
    "from genLabeledData import *\n",
    "\n",
    "gene_label_dict = {}\n",
    "with open(file_gene_path) as f:\n",
    "    for line in f.readlines():\n",
    "        arr = line.strip().split('\t')[1:]\n",
    "        for e in arr:\n",
    "            e = e.strip()\n",
    "            if e != '':\n",
    "                gene_label_dict[e] = 'Gene'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "520f77aa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['DEAD/H Box Polypeptide 11 Like 3',\n",
       " 'DEAD/H Box Polypeptide 11 Like 1',\n",
       " 'DEAD/H-Box Helicase 11 Like 1',\n",
       " 'DEAD/H (Asp-Glu-Ala-Asp/His) Box Polypeptide 11 Like 1',\n",
       " 'DDX11L3',\n",
       " 'DEAD/H (Asp-Glu-Ala-Asp/His) Box Helicase 11 Like 1',\n",
       " 'DDX11L9',\n",
       " 'DDX11L1',\n",
       " 'DEAD/H Box Polypeptide 11 Like 9',\n",
       " 'DEAD/H-Box Helicase 11 Like 1 (Pseudogene)',\n",
       " 'WAS Protein Family Homolog 7, Pseudogene',\n",
       " 'WASH5P',\n",
       " 'WASP Family Homolog 7, Pseudogene',\n",
       " 'WASH7P',\n",
       " 'WAS Protein Family Homolog 5 Pseudogene',\n",
       " 'Family With Sequence Similarity 39, Member F',\n",
       " 'FAM39F',\n",
       " 'Hsa-MiR-6859-5p',\n",
       " 'MI0022705',\n",
       " 'MIR6859-1',\n",
       " 'MIMAT0027618',\n",
       " 'Hsa-Mir-6859-1',\n",
       " 'MicroRNA Mir-6859-1',\n",
       " 'MicroRNA 6859-1',\n",
       " 'MIMAT0027619',\n",
       " 'MIR1302-2HG',\n",
       " 'Lnc-OR4F17-4',\n",
       " 'HSALNG0000005',\n",
       " 'NONHSAG024269.2',\n",
       " 'MIR1302-2 Host Gene',\n",
       " 'NONHSAG000003.2',\n",
       " 'MicroRNA 1302-2',\n",
       " 'MI0006363',\n",
       " 'Hsa-MiR-1302',\n",
       " 'Hsa-Mir-1302-2',\n",
       " 'RF00951',\n",
       " 'MIMAT0005890',\n",
       " 'MIRN1302-2',\n",
       " 'MIR1302-2',\n",
       " 'Chromosome 1 F379 Retina Specific Protein',\n",
       " 'F379',\n",
       " 'NONHSAG000004.2',\n",
       " 'FAM138F',\n",
       " 'HSALNG0000006',\n",
       " 'Family With Sequence Similarity 138 Member A',\n",
       " 'F379 Retina-Specific Protein',\n",
       " 'FAM138A',\n",
       " 'NONHSAG024270.2',\n",
       " 'Family With Sequence Similarity 138, Member A',\n",
       " 'HSALNG0122829',\n",
       " 'OR4G4P',\n",
       " 'Olfactory Receptor, Family 4, Subfamily G, Member 4 Pseudogene',\n",
       " 'Olfactory Receptor Family 4 Subfamily G Member 4 Pseudogene',\n",
       " 'Olfactory Receptor, Family 4, Subfamily G, Member 11 Pseudogene',\n",
       " 'Olfactory Receptor Family 4 Subfamily G Member 11 Pseudogene',\n",
       " 'OR4G11P',\n",
       " 'OR4F5',\n",
       " 'Olfactory Receptor, Family 4, Subfamily F, Member 5',\n",
       " 'Olfactory Receptor 4F5',\n",
       " 'Olfactory Receptor Family 4 Subfamily F Member 5',\n",
       " 'LINC00115',\n",
       " 'NONHSAG000029.2',\n",
       " 'NONHSAG000046.2',\n",
       " 'Lnc-OR4F29-1',\n",
       " 'HSALNG0000011',\n",
       " 'Novel Transcript',\n",
       " 'HSALNG0000040',\n",
       " 'ENSG00000238009',\n",
       " 'HSALNG0122836',\n",
       " 'AL627309.1',\n",
       " 'HSALNG0000022',\n",
       " 'NONHSAG000008.2',\n",
       " 'AL627309.3',\n",
       " 'ENSG00000239945',\n",
       " 'Capicua Transcriptional Repressor Pseudogene 27',\n",
       " 'Capicua Homolog Pseudogene',\n",
       " 'CICP27',\n",
       " 'AL627309.6',\n",
       " 'Novel Pseudogene',\n",
       " 'ENSG00000268903',\n",
       " 'ENSG00000269981',\n",
       " 'AL627309.7',\n",
       " 'AL627309.2',\n",
       " 'Lnc-OR4F29-7',\n",
       " 'NONHSAG000011.2',\n",
       " 'ENSG00000239906',\n",
       " 'ENSG00000241860',\n",
       " 'HSALNG0000014',\n",
       " 'AL627309.5',\n",
       " 'Lnc-OR4F29-3',\n",
       " 'NONHSAG000012.2',\n",
       " 'RNA, U6 Small Nuclear 1100, Pseudogene',\n",
       " 'RNU6-1100P',\n",
       " 'Lnc-OR4F5-2',\n",
       " 'ENSG00000241599',\n",
       " 'NONHSAG000014.2',\n",
       " 'HSALNG0000016',\n",
       " 'AL627309.4',\n",
       " 'Putative ATP-Dependent RNA Helicase DDX12',\n",
       " 'DEAD/H-Box Helicase 11 Like 17 (Pseudogene)']"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arr = [k for k, v in gene_label_dict.items()]\n",
    "arr[:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "d31ba89c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "processing line: 1\n",
      "time cost: 50.86302638053894\n",
      "processing line: 2\n",
      "time cost: 55.17097020149231\n",
      "processing line: 3\n",
      "time cost: 56.064029932022095\n",
      "processing line: 4\n",
      "time cost: 57.75397181510925\n",
      "processing line: 5\n",
      "time cost: 55.39799928665161\n",
      "processing line: 6\n",
      "time cost: 60.985029458999634\n",
      "processing line: 7\n",
      "time cost: 56.05196928977966\n",
      "processing line: 8\n",
      "time cost: 43.34500050544739\n",
      "processing line: 9\n",
      "time cost: 64.72300124168396\n",
      "processing line: 10\n",
      "time cost: 60.65499949455261\n",
      "processing line: 11\n",
      "time cost: 49.19297003746033\n",
      "processing line: 12\n",
      "time cost: 55.74174952507019\n",
      "processing line: 13\n",
      "time cost: 53.07337498664856\n",
      "processing line: 14\n",
      "time cost: 47.73800301551819\n",
      "processing line: 15\n",
      "time cost: 84.88799834251404\n",
      "processing line: 16\n",
      "time cost: 48.14700174331665\n",
      "processing line: 17\n",
      "time cost: 52.77799582481384\n",
      "processing line: 18\n",
      "time cost: 38.22300362586975\n",
      "processing line: 19\n",
      "time cost: 49.215970516204834\n",
      "processing line: 20\n",
      "time cost: 53.02002954483032\n",
      "processing line: 21\n",
      "time cost: 46.84696912765503\n",
      "processing line: 22\n",
      "time cost: 53.84303140640259\n",
      "processing line: 23\n",
      "time cost: 56.13997149467468\n",
      "processing line: 24\n",
      "time cost: 48.960028409957886\n",
      "processing line: 25\n",
      "time cost: 58.51148724555969\n",
      "processing line: 26\n",
      "time cost: 51.70802974700928\n",
      "processing line: 27\n",
      "time cost: 57.67395758628845\n",
      "processing line: 28\n",
      "time cost: 52.139029026031494\n",
      "processing line: 29\n",
      "time cost: 51.29299974441528\n",
      "processing line: 30\n",
      "time cost: 55.52500605583191\n",
      "processing line: 31\n",
      "time cost: 38.477997064590454\n",
      "processing line: 32\n",
      "time cost: 55.08899760246277\n",
      "processing line: 33\n",
      "time cost: 46.11600112915039\n",
      "processing line: 34\n",
      "time cost: 49.13600134849548\n",
      "processing line: 35\n",
      "time cost: 46.03803014755249\n",
      "processing line: 36\n",
      "time cost: 49.3649697303772\n",
      "processing line: 37\n",
      "time cost: 52.11499834060669\n",
      "processing line: 38\n",
      "time cost: 48.9709997177124\n",
      "processing line: 39\n",
      "time cost: 52.4260311126709\n",
      "processing line: 40\n",
      "time cost: 62.586039304733276\n",
      "processing line: 41\n",
      "time cost: 50.298996686935425\n",
      "processing line: 42\n",
      "time cost: 54.421003103256226\n",
      "processing line: 43\n",
      "time cost: 48.98596739768982\n",
      "processing line: 44\n",
      "time cost: 46.65002942085266\n",
      "processing line: 45\n",
      "time cost: 52.16221809387207\n",
      "processing line: 46\n",
      "time cost: 48.50996685028076\n",
      "processing line: 47\n",
      "time cost: 53.16699981689453\n",
      "processing line: 48\n",
      "time cost: 58.67306685447693\n",
      "processing line: 49\n",
      "time cost: 52.689013957977295\n",
      "processing line: 50\n",
      "time cost: 48.746992111206055\n",
      "processing line: 51\n",
      "time cost: 59.534972190856934\n",
      "processing line: 52\n",
      "time cost: 52.40202879905701\n",
      "processing line: 53\n",
      "time cost: 47.82397150993347\n",
      "processing line: 54\n",
      "time cost: 49.47184133529663\n",
      "processing line: 55\n",
      "time cost: 61.75470757484436\n",
      "processing line: 56\n",
      "time cost: 50.35097026824951\n",
      "processing line: 57\n",
      "time cost: 47.87702941894531\n",
      "processing line: 58\n",
      "time cost: 53.175968408584595\n",
      "processing line: 59\n",
      "time cost: 45.927029609680176\n",
      "processing line: 60\n",
      "time cost: 55.73397207260132\n",
      "processing line: 61\n",
      "time cost: 45.457998514175415\n",
      "processing line: 62\n",
      "time cost: 50.37203097343445\n",
      "processing line: 63\n",
      "time cost: 57.81699800491333\n",
      "processing line: 64\n",
      "time cost: 54.275001764297485\n",
      "processing line: 65\n",
      "time cost: 54.40399980545044\n",
      "processing line: 66\n",
      "time cost: 52.02399969100952\n",
      "processing line: 67\n",
      "time cost: 53.64999723434448\n",
      "processing line: 68\n",
      "time cost: 62.993003368377686\n",
      "processing line: 69\n",
      "time cost: 58.640100717544556\n",
      "processing line: 70\n",
      "time cost: 54.31781840324402\n",
      "processing line: 71\n",
      "time cost: 46.750027894973755\n",
      "processing line: 72\n",
      "time cost: 49.913002014160156\n",
      "processing line: 73\n",
      "time cost: 65.05999827384949\n",
      "processing line: 74\n",
      "time cost: 59.00599956512451\n",
      "processing line: 75\n",
      "time cost: 55.06096959114075\n",
      "processing line: 76\n",
      "time cost: 46.702001333236694\n",
      "processing line: 77\n",
      "time cost: 46.40802717208862\n",
      "processing line: 78\n",
      "time cost: 50.974377393722534\n",
      "processing line: 79\n",
      "time cost: 62.01596164703369\n",
      "processing line: 80\n",
      "time cost: 51.08000111579895\n",
      "processing line: 81\n",
      "time cost: 50.94300031661987\n",
      "processing line: 82\n",
      "time cost: 53.378000020980835\n",
      "processing line: 83\n",
      "time cost: 41.28099966049194\n",
      "processing line: 84\n",
      "time cost: 53.79200005531311\n",
      "processing line: 85\n",
      "time cost: 51.495004177093506\n",
      "processing line: 86\n",
      "time cost: 51.687968254089355\n",
      "processing line: 87\n",
      "time cost: 44.4550302028656\n",
      "processing line: 88\n",
      "time cost: 50.117966651916504\n",
      "processing line: 89\n",
      "time cost: 51.46103119850159\n",
      "processing line: 90\n",
      "time cost: 51.80400347709656\n",
      "processing line: 91\n",
      "time cost: 42.79796838760376\n",
      "processing line: 92\n",
      "time cost: 50.955132484436035\n",
      "processing line: 93\n",
      "time cost: 51.18397045135498\n",
      "processing line: 94\n",
      "time cost: 37.98602771759033\n",
      "processing line: 95\n",
      "time cost: 50.01900053024292\n",
      "processing line: 96\n",
      "time cost: 51.506324768066406\n",
      "processing line: 97\n",
      "time cost: 57.590898752212524\n",
      "processing line: 98\n",
      "time cost: 51.514031171798706\n",
      "processing line: 99\n",
      "time cost: 58.81397104263306\n",
      "processing line: 100\n",
      "time cost: 49.49499750137329\n",
      "processing line: 101\n",
      "time cost: 47.278000831604004\n",
      "processing line: 102\n",
      "time cost: 53.916999101638794\n",
      "processing line: 103\n",
      "time cost: 45.11699962615967\n",
      "processing line: 104\n",
      "time cost: 52.20900344848633\n",
      "processing line: 105\n",
      "time cost: 66.94905662536621\n",
      "processing line: 106\n",
      "time cost: 49.31699991226196\n",
      "processing line: 107\n",
      "time cost: 66.29816699028015\n",
      "processing line: 108\n",
      "time cost: 46.889002084732056\n",
      "processing line: 109\n",
      "time cost: 49.11602830886841\n",
      "processing line: 110\n",
      "time cost: 46.75296974182129\n",
      "processing line: 111\n",
      "time cost: 49.962010622024536\n",
      "processing line: 112\n",
      "time cost: 42.20403838157654\n",
      "processing line: 113\n",
      "time cost: 48.9399676322937\n",
      "processing line: 114\n",
      "time cost: 47.866997957229614\n",
      "processing line: 115\n",
      "time cost: 48.66203856468201\n",
      "processing line: 116\n",
      "time cost: 54.69402837753296\n",
      "processing line: 117\n",
      "time cost: 58.63600015640259\n",
      "processing line: 118\n",
      "time cost: 57.71299958229065\n",
      "processing line: 119\n",
      "time cost: 55.18597221374512\n",
      "processing line: 120\n",
      "time cost: 56.31299710273743\n",
      "processing line: 121\n",
      "time cost: 53.57300114631653\n",
      "processing line: 122\n",
      "time cost: 53.55402946472168\n",
      "processing line: 123\n",
      "time cost: 45.02601218223572\n",
      "processing line: 124\n",
      "time cost: 52.71696972846985\n",
      "processing line: 125\n",
      "time cost: 55.816999673843384\n",
      "processing line: 126\n",
      "time cost: 62.11282205581665\n",
      "processing line: 127\n",
      "time cost: 52.40103626251221\n",
      "processing line: 128\n",
      "time cost: 53.22516584396362\n",
      "processing line: 129\n",
      "time cost: 50.25396966934204\n",
      "processing line: 130\n",
      "time cost: 56.26563549041748\n",
      "processing line: 131\n",
      "time cost: 46.76399874687195\n",
      "processing line: 132\n",
      "time cost: 48.42091226577759\n",
      "processing line: 133\n",
      "time cost: 54.48499917984009\n",
      "processing line: 134\n",
      "time cost: 58.22697162628174\n",
      "processing line: 135\n",
      "time cost: 55.72603416442871\n",
      "processing line: 136\n",
      "time cost: 50.46797156333923\n",
      "processing line: 137\n",
      "time cost: 56.51103138923645\n",
      "processing line: 138\n",
      "time cost: 54.34999942779541\n",
      "processing line: 139\n",
      "time cost: 49.138004302978516\n",
      "processing line: 140\n",
      "time cost: 51.12999486923218\n",
      "processing line: 141\n",
      "time cost: 49.66697645187378\n",
      "processing line: 142\n",
      "time cost: 50.615002155303955\n",
      "processing line: 143\n",
      "time cost: 60.742000102996826\n",
      "processing line: 144\n",
      "time cost: 50.996999979019165\n",
      "processing line: 145\n",
      "time cost: 49.459033250808716\n",
      "processing line: 146\n",
      "time cost: 49.16404128074646\n",
      "processing line: 147\n",
      "time cost: 58.414008378982544\n",
      "processing line: 148\n",
      "time cost: 50.98399114608765\n",
      "processing line: 149\n",
      "time cost: 48.25710606575012\n",
      "processing line: 150\n",
      "time cost: 44.52600169181824\n",
      "processing line: 151\n",
      "time cost: 49.068997621536255\n",
      "processing line: 152\n",
      "time cost: 49.92899990081787\n",
      "processing line: 153\n",
      "time cost: 62.62399935722351\n",
      "processing line: 154\n",
      "time cost: 57.315998792648315\n",
      "processing line: 155\n",
      "time cost: 54.992825984954834\n",
      "processing line: 156\n",
      "time cost: 47.85300660133362\n",
      "processing line: 157\n",
      "time cost: 49.85001540184021\n",
      "processing line: 158\n",
      "time cost: 58.473681926727295\n",
      "processing line: 159\n",
      "time cost: 71.03402996063232\n",
      "processing line: 160\n",
      "time cost: 51.91597008705139\n",
      "processing line: 161\n",
      "time cost: 49.73933434486389\n",
      "processing line: 162\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Input \u001b[1;32mIn [6]\u001b[0m, in \u001b[0;36m<cell line: 2>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m test_text_data \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m'\u001b[39m\u001b[38;5;130;01m\\n\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;241m.\u001b[39mjoin(item) \u001b[38;5;28;01mfor\u001b[39;00m item \u001b[38;5;129;01min\u001b[39;00m test_data]\n\u001b[1;32m----> 2\u001b[0m test_json_data \u001b[38;5;241m=\u001b[39m \u001b[43mprocess_by_regex\u001b[49m\u001b[43m(\u001b[49m\u001b[43mgene_label_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtest_text_data\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlanguage\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mEN\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m      4\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m \u001b[38;5;28mopen\u001b[39m(\u001b[38;5;124mr\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mtest_data.json\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mw\u001b[39m\u001b[38;5;124m\"\u001b[39m) \u001b[38;5;28;01mas\u001b[39;00m f:\n\u001b[0;32m      5\u001b[0m     f\u001b[38;5;241m.\u001b[39mwrite(json\u001b[38;5;241m.\u001b[39mdumps(test_json_data, indent\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m1\u001b[39m, separators\u001b[38;5;241m=\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m,\u001b[39m\u001b[38;5;124m'\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124m:\u001b[39m\u001b[38;5;124m'\u001b[39m), ensure_ascii\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m))\n",
      "File \u001b[1;32mD:\\projects\\NLP\\Gene\\genLabeledData.py:95\u001b[0m, in \u001b[0;36mprocess_by_regex\u001b[1;34m(text_label_dict, texts, language)\u001b[0m\n\u001b[0;32m     93\u001b[0m \u001b[38;5;28mprint\u001b[39m(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mprocessing line: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m%\u001b[39m (line_no\u001b[38;5;241m+\u001b[39m\u001b[38;5;241m1\u001b[39m))\n\u001b[0;32m     94\u001b[0m text \u001b[38;5;241m=\u001b[39m text\u001b[38;5;241m.\u001b[39mstrip()\n\u001b[1;32m---> 95\u001b[0m labels \u001b[38;5;241m=\u001b[39m \u001b[43mlabel_line_by_regex\u001b[49m\u001b[43m(\u001b[49m\u001b[43mtext\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext_label_dict\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlanguage\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     96\u001b[0m entities \u001b[38;5;241m=\u001b[39m []\n\u001b[0;32m     97\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m \u001b[38;5;28mid\u001b[39m, start, end, word, label \u001b[38;5;129;01min\u001b[39;00m labels:\n",
      "File \u001b[1;32mD:\\projects\\NLP\\Gene\\genLabeledData.py:68\u001b[0m, in \u001b[0;36mlabel_line_by_regex\u001b[1;34m(text, text_label_dict, language)\u001b[0m\n\u001b[0;32m     66\u001b[0m     matches \u001b[38;5;241m=\u001b[39m re\u001b[38;5;241m.\u001b[39mfinditer(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m(^| )\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m( |$|\u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124m.|,)\u001b[39m\u001b[38;5;124m'\u001b[39m \u001b[38;5;241m%\u001b[39m key, text, re\u001b[38;5;241m.\u001b[39mI)\n\u001b[0;32m     67\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[1;32m---> 68\u001b[0m     matches \u001b[38;5;241m=\u001b[39m \u001b[43mre\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mfinditer\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43m(^| )\u001b[39;49m\u001b[38;5;132;43;01m%s\u001b[39;49;00m\u001b[38;5;124;43m( |$|\u001b[39;49m\u001b[38;5;124;43m\\\u001b[39;49m\u001b[38;5;124;43m.|,)\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m%\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mkey\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtext\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     70\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m match \u001b[38;5;129;01min\u001b[39;00m matches:\n\u001b[0;32m     71\u001b[0m     start, end \u001b[38;5;241m=\u001b[39m match\u001b[38;5;241m.\u001b[39mspan()\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\re.py:248\u001b[0m, in \u001b[0;36mfinditer\u001b[1;34m(pattern, string, flags)\u001b[0m\n\u001b[0;32m    243\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mfinditer\u001b[39m(pattern, string, flags\u001b[38;5;241m=\u001b[39m\u001b[38;5;241m0\u001b[39m):\n\u001b[0;32m    244\u001b[0m     \u001b[38;5;124;03m\"\"\"Return an iterator over all non-overlapping matches in the\u001b[39;00m\n\u001b[0;32m    245\u001b[0m \u001b[38;5;124;03m    string.  For each match, the iterator returns a Match object.\u001b[39;00m\n\u001b[0;32m    246\u001b[0m \n\u001b[0;32m    247\u001b[0m \u001b[38;5;124;03m    Empty matches are included in the result.\"\"\"\u001b[39;00m\n\u001b[1;32m--> 248\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_compile\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpattern\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mflags\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mfinditer(string)\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\re.py:304\u001b[0m, in \u001b[0;36m_compile\u001b[1;34m(pattern, flags)\u001b[0m\n\u001b[0;32m    302\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m sre_compile\u001b[38;5;241m.\u001b[39misstring(pattern):\n\u001b[0;32m    303\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mTypeError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mfirst argument must be string or compiled pattern\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m--> 304\u001b[0m p \u001b[38;5;241m=\u001b[39m \u001b[43msre_compile\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mcompile\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpattern\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mflags\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    305\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m (flags \u001b[38;5;241m&\u001b[39m DEBUG):\n\u001b[0;32m    306\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(_cache) \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m _MAXCACHE:\n\u001b[0;32m    307\u001b[0m         \u001b[38;5;66;03m# Drop the oldest item\u001b[39;00m\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_compile.py:764\u001b[0m, in \u001b[0;36mcompile\u001b[1;34m(p, flags)\u001b[0m\n\u001b[0;32m    762\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m isstring(p):\n\u001b[0;32m    763\u001b[0m     pattern \u001b[38;5;241m=\u001b[39m p\n\u001b[1;32m--> 764\u001b[0m     p \u001b[38;5;241m=\u001b[39m \u001b[43msre_parse\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mparse\u001b[49m\u001b[43m(\u001b[49m\u001b[43mp\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mflags\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    765\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    766\u001b[0m     pattern \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_parse.py:950\u001b[0m, in \u001b[0;36mparse\u001b[1;34m(str, flags, state)\u001b[0m\n\u001b[0;32m    947\u001b[0m state\u001b[38;5;241m.\u001b[39mstr \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m\n\u001b[0;32m    949\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 950\u001b[0m     p \u001b[38;5;241m=\u001b[39m \u001b[43m_parse_sub\u001b[49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mflags\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m&\u001b[39;49m\u001b[43m \u001b[49m\u001b[43mSRE_FLAG_VERBOSE\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m0\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m    951\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m Verbose:\n\u001b[0;32m    952\u001b[0m     \u001b[38;5;66;03m# the VERBOSE flag was switched on inside the pattern.  to be\u001b[39;00m\n\u001b[0;32m    953\u001b[0m     \u001b[38;5;66;03m# on the safe side, we'll parse the whole thing again...\u001b[39;00m\n\u001b[0;32m    954\u001b[0m     state \u001b[38;5;241m=\u001b[39m State()\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_parse.py:443\u001b[0m, in \u001b[0;36m_parse_sub\u001b[1;34m(source, state, verbose, nested)\u001b[0m\n\u001b[0;32m    441\u001b[0m start \u001b[38;5;241m=\u001b[39m source\u001b[38;5;241m.\u001b[39mtell()\n\u001b[0;32m    442\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m--> 443\u001b[0m     itemsappend(\u001b[43m_parse\u001b[49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnested\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    444\u001b[0m \u001b[43m                       \u001b[49m\u001b[38;5;129;43;01mnot\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mnested\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mand\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mnot\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mitems\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[0;32m    445\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m sourcematch(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m|\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m    446\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_parse.py:836\u001b[0m, in \u001b[0;36m_parse\u001b[1;34m(source, state, verbose, nested, first)\u001b[0m\n\u001b[0;32m    833\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m source\u001b[38;5;241m.\u001b[39merror(err\u001b[38;5;241m.\u001b[39mmsg, \u001b[38;5;28mlen\u001b[39m(name) \u001b[38;5;241m+\u001b[39m \u001b[38;5;241m1\u001b[39m) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;28mNone\u001b[39m\n\u001b[0;32m    834\u001b[0m sub_verbose \u001b[38;5;241m=\u001b[39m ((verbose \u001b[38;5;129;01mor\u001b[39;00m (add_flags \u001b[38;5;241m&\u001b[39m SRE_FLAG_VERBOSE)) \u001b[38;5;129;01mand\u001b[39;00m\n\u001b[0;32m    835\u001b[0m                \u001b[38;5;129;01mnot\u001b[39;00m (del_flags \u001b[38;5;241m&\u001b[39m SRE_FLAG_VERBOSE))\n\u001b[1;32m--> 836\u001b[0m p \u001b[38;5;241m=\u001b[39m \u001b[43m_parse_sub\u001b[49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43msub_verbose\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnested\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[0;32m    837\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m source\u001b[38;5;241m.\u001b[39mmatch(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m)\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m    838\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m source\u001b[38;5;241m.\u001b[39merror(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmissing ), unterminated subpattern\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m    839\u001b[0m                        source\u001b[38;5;241m.\u001b[39mtell() \u001b[38;5;241m-\u001b[39m start)\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_parse.py:443\u001b[0m, in \u001b[0;36m_parse_sub\u001b[1;34m(source, state, verbose, nested)\u001b[0m\n\u001b[0;32m    441\u001b[0m start \u001b[38;5;241m=\u001b[39m source\u001b[38;5;241m.\u001b[39mtell()\n\u001b[0;32m    442\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m--> 443\u001b[0m     itemsappend(\u001b[43m_parse\u001b[49m\u001b[43m(\u001b[49m\u001b[43msource\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstate\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mverbose\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnested\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m+\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;241;43m1\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    444\u001b[0m \u001b[43m                       \u001b[49m\u001b[38;5;129;43;01mnot\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mnested\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mand\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[38;5;129;43;01mnot\u001b[39;49;00m\u001b[43m \u001b[49m\u001b[43mitems\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[0;32m    445\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m sourcematch(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m|\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n\u001b[0;32m    446\u001b[0m         \u001b[38;5;28;01mbreak\u001b[39;00m\n",
      "File \u001b[1;32mD:\\Anaconda3\\lib\\sre_parse.py:844\u001b[0m, in \u001b[0;36m_parse\u001b[1;34m(source, state, verbose, nested, first)\u001b[0m\n\u001b[0;32m    841\u001b[0m         state\u001b[38;5;241m.\u001b[39mclosegroup(group, p)\n\u001b[0;32m    842\u001b[0m     subpatternappend((SUBPATTERN, (group, add_flags, del_flags, p)))\n\u001b[1;32m--> 844\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[43mthis\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m==\u001b[39;49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m^\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m:\n\u001b[0;32m    845\u001b[0m     subpatternappend((AT, AT_BEGINNING))\n\u001b[0;32m    847\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m this \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m$\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "test_text_data = ['\\n'.join(item) for item in test_data]\n",
    "test_json_data = process_by_regex(gene_label_dict, test_text_data, language='EN')\n",
    "\n",
    "with open(r'test_data.json', \"w\") as f:\n",
    "    f.write(json.dumps(test_json_data, indent=1, separators=(',', ':'), ensure_ascii=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1083bafb",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "train_text_data = ['\\n'.join(item) for item in train_data]\n",
    "train_json_data = process_by_regex(gene_label_dict, train_text_data, language='EN')\n",
    "\n",
    "with open(r'train_data.json', \"w\") as f:\n",
    "    f.write(json.dumps(train_json_data, indent=1, separators=(',', ':'), ensure_ascii=False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d35e8c87",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f37bee15",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9febfd42",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "46c0dce6",
   "metadata": {},
   "outputs": [],
   "source": [
    "### 文本转化为句子\n",
    "data_all = data_wnt4 + data_ca\n",
    "data_cls = []\n",
    "for item in data_all:\n",
    "    item_sent = []\n",
    "    for line in item:\n",
    "        arr = line.split('. ')\n",
    "        for sent in arr:\n",
    "            sent = sent.strip()\n",
    "            if sent != '':\n",
    "                if sent[-1] != '.':\n",
    "                    item_sent.append(sent + '.')\n",
    "                else:\n",
    "                    item_sent.append(sent)\n",
    "    data_cls.append(item_sent)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "279f2994",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(r'D:\\项目资料\\基因表达\\通路分析\\gene_data_sentences.txt', 'w') as f:\n",
    "    for item in data_cls:\n",
    "        for sent in item:\n",
    "            f.write('%s\\n' % sent)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d53dd89b",
   "metadata": {},
   "source": [
    "### 选择基因"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "164aabe0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37487\n"
     ]
    }
   ],
   "source": [
    "word_list = []\n",
    "for item in data_all:\n",
    "    item_str = ' '.join(item)\n",
    "    word_arr = item_str.replace(',', '').replace('.', '').split(' ')\n",
    "    word_list.extend(list(set(word_arr)))\n",
    "word_list = list(set(word_list))\n",
    "print(len(word_list))\n",
    "word_str = ' '.join(word_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "3c334cc7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<re.Match object; span=(197401, 197408), match=' NDRG1 '>"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gr = '([ ]|^)' + 'NDRG1'.replace('.', '\\.').replace('-', '\\-') + '([ ,]|$)'\n",
    "s = 'Rapid detection of the hypoxia-regulated CA-IX and NDRG1 gene expression in different glioblastoma cells in vitro.'\n",
    "re.search(gr, word_str, re.I)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "0b7efe70",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "100\n",
      "DVL1 in\n",
      "PRDM16 in\n",
      "200\n",
      "300\n",
      "ENO1 in\n",
      "CA6 in\n",
      "MIR34A in\n",
      "CTNNBIP1 in\n",
      "MTOR in\n",
      "400\n",
      "500\n",
      "SDHB in\n",
      "600\n",
      "PADI4 in\n",
      "700\n",
      "WNT4 in\n",
      "EPHB2 in\n",
      "800\n",
      "RHD in\n",
      "ARID1A in\n",
      "SFN in\n",
      "900\n",
      "1000\n",
      "HPCA in\n",
      "1100\n",
      "GJB5 in\n",
      "1200\n",
      "RSPO1 in\n",
      "1300\n",
      "SLC2A1 in\n",
      "CDC20 in\n",
      "1400\n",
      "PTPRF in\n",
      "1500\n",
      "CYP4B1 in\n",
      "1600\n",
      "LRP8 in\n",
      "1700\n",
      "C8B in\n",
      "JUN in\n",
      "1800\n",
      "1900\n",
      "WLS in\n",
      "2000\n",
      "DNAJB4 in\n",
      "2100\n",
      "2200\n",
      "2300\n",
      "F3 in\n",
      "2400\n",
      "COL11A1 in\n",
      "2500\n",
      "MYBPHL in\n",
      "GSTM1 in\n",
      "CSF1 in\n",
      "2600\n",
      "WNT2B in\n",
      "RHOC in\n",
      "2700\n",
      "NRAS in\n",
      "NGF in\n",
      "2800\n",
      "HSD3B1 in\n",
      "NOTCH2 in\n",
      "2900\n",
      "PIAS3 in\n",
      "TXNIP in\n",
      "3000\n",
      "3100\n",
      "3200\n",
      "S100A10 in\n",
      "S100A11 in\n",
      "CRNN in\n",
      "S100A9 in\n",
      "S100A6 in\n",
      "3300\n",
      "S100A4 in\n",
      "S100A1 in\n",
      "RAB13 in\n",
      "SHE in\n",
      "3400\n",
      "MUC1 in\n",
      "SYT11 in\n",
      "RAB25 in\n",
      "NES in\n",
      "HDGF in\n",
      "3500\n",
      "PRCC in\n",
      "CD1D in\n",
      "AIM2 in\n",
      "3600\n",
      "CRP in\n",
      "SDHC in\n",
      "3700\n",
      "RGS5 in\n",
      "3800\n",
      "F5 in\n",
      "KIFAP3 in\n",
      "3900\n",
      "4000\n",
      "TNR in\n",
      "4100\n",
      "LAMC2 in\n",
      "4200\n",
      "PDC in\n",
      "4300\n",
      "LHX9 in\n",
      "ELF3 in\n",
      "4400\n",
      "REN in\n",
      "4500\n",
      "CD55 in\n",
      "CD46 in\n",
      "CD34 in\n",
      "4600\n",
      "HSD11B1 in\n",
      "IRF6 in\n",
      "ATF3 in\n",
      "4700\n",
      "TGFB2 in\n",
      "4800\n",
      "ENAH in\n",
      "LEFTY2 in\n",
      "4900\n",
      "WNT9A in\n",
      "WNT3A in\n",
      "5000\n",
      "RHOU in\n",
      "RAB4A in\n",
      "AGT in\n",
      "5100\n",
      "5200\n",
      "FH in\n",
      "AKT3 in\n",
      "5300\n",
      "NLRP3 in\n",
      "5400\n",
      "5500\n",
      "5600\n",
      "RRM2 in\n",
      "ROCK2 in\n",
      "GREB1 in\n",
      "5700\n",
      "OSR1 in\n",
      "SDC1 in\n",
      "GDF7 in\n",
      "5800\n",
      "5900\n",
      "CAD in\n",
      "ALK in\n",
      "6000\n",
      "XDH in\n",
      "6100\n",
      "MTA3 in\n",
      "6200\n",
      "SIX3 in\n",
      "SIX2 in\n",
      "EPCAM in\n",
      "MSH2 in\n",
      "MSH6 in\n",
      "6300\n",
      "FSHR in\n",
      "EFEMP1 in\n",
      "6400\n",
      "XPO1 in\n",
      "6500\n",
      "6600\n",
      "FIGLA in\n",
      "6700\n",
      "HK2 in\n",
      "6800\n",
      "TCF7L1 in\n",
      "6900\n",
      "7000\n",
      "7100\n",
      "7200\n",
      "TSGA10 in\n",
      "7300\n",
      "POU3F3 in\n",
      "7400\n",
      "7500\n",
      "PAX8 in\n",
      "FAM138B in\n",
      "7600\n",
      "INHBB in\n",
      "GLI2 in\n",
      "7700\n",
      "7800\n",
      "7900\n",
      "CXCR4 in\n",
      "8000\n",
      "LRP1B in\n",
      "ZEB2 in\n",
      "8100\n",
      "KIF5C in\n",
      "TNFAIP6 in\n",
      "NR4A2 in\n",
      "8200\n",
      "FAP in\n",
      "8300\n",
      "LRP2 in\n",
      "DLX2 in\n",
      "8400\n",
      "PDK1 in\n",
      "OLA1 in\n",
      "HOXD11 in\n",
      "8500\n",
      "NFE2L2 in\n",
      "FRZB in\n",
      "8600\n",
      "GLS in\n",
      "STAT1 in\n",
      "8700\n",
      "PCGEM1 in\n",
      "8800\n",
      "CLK1 in\n",
      "8900\n",
      "CD28 in\n",
      "CTLA4 in\n",
      "CPO in\n",
      "FZD5 in\n",
      "9000\n",
      "IDH1 in\n",
      "ERBB4 in\n",
      "ATIC in\n",
      "FN1 in\n",
      "9100\n",
      "CXCR2 in\n",
      "WNT6 in\n",
      "WNT10A in\n",
      "9200\n",
      "DES in\n",
      "INHA in\n",
      "9300\n",
      "COL4A4 in\n",
      "CCL20 in\n",
      "9400\n",
      "INPP5D in\n",
      "9500\n",
      "9600\n",
      "9700\n",
      "9800\n",
      "VHL in\n",
      "WNT7A in\n",
      "9900\n",
      "EAF1 in\n",
      "10000\n",
      "10100\n",
      "MLH1 in\n",
      "10200\n",
      "CTNNB1 in\n",
      "CCK in\n",
      "10300\n",
      "CCR1 in\n",
      "10400\n",
      "CDC25A in\n",
      "CAMP in\n",
      "SLC26A6 in\n",
      "10500\n",
      "GPX1 in\n",
      "RHOA in\n",
      "CAMKV in\n",
      "RASSF1 in\n",
      "10600\n",
      "TLR9 in\n",
      "BAP1 in\n",
      "STAB1 in\n",
      "PBRM1 in\n",
      "ITIH4 in\n",
      "PRKCD in\n",
      "10700\n",
      "WNT5A in\n",
      "FHIT in\n",
      "PTPRG in\n",
      "10800\n",
      "MITF in\n",
      "10900\n",
      "11000\n",
      "11100\n",
      "11200\n",
      "CD47 in\n",
      "11300\n",
      "11400\n",
      "CD80 in\n",
      "GSK3B in\n",
      "CD86 in\n",
      "11500\n",
      "11600\n",
      "RHO in\n",
      "11700\n",
      "11800\n",
      "FOXL2 in\n",
      "RBP1 in\n",
      "ATR in\n",
      "11900\n",
      "CP in\n",
      "WWTR1 in\n",
      "12000\n",
      "GPR87 in\n",
      "12100\n",
      "SCHIP1 in\n",
      "12200\n",
      "SI in\n",
      "BCHE in\n",
      "12300\n",
      "12400\n",
      "ZMAT3 in\n",
      "PIK3CA in\n",
      "SOX2 in\n",
      "12500\n",
      "DVL3 in\n",
      "12600\n",
      "SST in\n",
      "FGF12 in\n",
      "12700\n",
      "LINC00887 in\n",
      "MUC4 in\n",
      "TFRC in\n",
      "12800\n",
      "12900\n",
      "FGFR3 in\n",
      "13000\n",
      "PPP2R2C in\n",
      "S100P in\n",
      "13100\n",
      "13200\n",
      "PROM1 in\n",
      "13300\n",
      "SLC34A2 in\n",
      "13400\n",
      "RFC1 in\n",
      "13500\n",
      "TEC in\n",
      "13600\n",
      "PDGFRA in\n",
      "KIT in\n",
      "KDR in\n",
      "CLOCK in\n",
      "13700\n",
      "REST in\n",
      "13800\n",
      "SULT1E1 in\n",
      "13900\n",
      "MUC7 in\n",
      "MOB1B in\n",
      "SLC4A4 in\n",
      "GC in\n",
      "AFP in\n",
      "AREG in\n",
      "CXCL9 in\n",
      "CXCL10 in\n",
      "14000\n",
      "CXCL13 in\n",
      "14100\n",
      "SPP1 in\n",
      "14200\n",
      "14300\n",
      "14400\n",
      "LEF1 in\n",
      "EGF in\n",
      "14500\n",
      "14600\n",
      "IL2 in\n",
      "FGF2 in\n",
      "SPRY1 in\n",
      "14700\n",
      "SLC7A11 in\n",
      "14800\n",
      "MAML3 in\n",
      "14900\n",
      "15000\n",
      "TLR2 in\n",
      "SFRP2 in\n",
      "15100\n",
      "15200\n",
      "HAND2 in\n",
      "15300\n",
      "15400\n",
      "TLR3 in\n",
      "15500\n",
      "SDHA in\n",
      "TERT in\n",
      "15600\n",
      "MTRR in\n",
      "15700\n",
      "15800\n",
      "15900\n",
      "16000\n",
      "NPR3 in\n",
      "AMACR in\n",
      "PRLR in\n",
      "16100\n",
      "GDNF in\n",
      "C9 in\n",
      "C7 in\n",
      "C6 in\n",
      "16200\n",
      "FGF10 in\n",
      "ESM1 in\n",
      "16300\n",
      "GZMA in\n",
      "MAP3K1 in\n",
      "16400\n",
      "PIK3R1 in\n",
      "16500\n",
      "CCNB1 in\n",
      "TNPO1 in\n",
      "16600\n",
      "16700\n",
      "DHFR in\n",
      "SSBP2 in\n",
      "16800\n",
      "16900\n",
      "NR2F1 in\n",
      "CAST in\n",
      "17000\n",
      "17100\n",
      "APC in\n",
      "MCC in\n",
      "ATG12 in\n",
      "17200\n",
      "LOX in\n",
      "17300\n",
      "17400\n",
      "GDF9 in\n",
      "TCF7 in\n",
      "PITX1 in\n",
      "17500\n",
      "WNT8A in\n",
      "17600\n",
      "SLC4A9 in\n",
      "17700\n",
      "17800\n",
      "17900\n",
      "PDGFRB in\n",
      "CAMK2A in\n",
      "SPARC in\n",
      "18000\n",
      "EBF1 in\n",
      "18100\n",
      "FOXI1 in\n",
      "18200\n",
      "18300\n",
      "FGFR4 in\n",
      "PROP1 in\n",
      "18400\n",
      "RACK1 in\n",
      "18500\n",
      "IRF4 in\n",
      "18600\n",
      "BMP6 in\n",
      "18700\n",
      "EDN1 in\n",
      "18800\n",
      "PRL in\n",
      "18900\n",
      "19000\n",
      "19100\n",
      "MICE in\n",
      "19200\n",
      "19300\n",
      "TNF in\n",
      "C2 in\n",
      "19400\n",
      "AGER in\n",
      "NOTCH4 in\n",
      "19500\n",
      "SRPK1 in\n",
      "ETV7 in\n",
      "19600\n",
      "CDKN1A in\n",
      "19700\n",
      "TFEB in\n",
      "SRF in\n",
      "19800\n",
      "VEGFA in\n",
      "RUNX2 in\n",
      "CD2AP in\n",
      "19900\n",
      "20000\n",
      "20100\n",
      "20200\n",
      "20300\n",
      "20400\n",
      "20500\n",
      "CD24 in\n",
      "SOBP in\n",
      "FOXO3 in\n",
      "20600\n",
      "20700\n",
      "20800\n",
      "RSPO3 in\n",
      "ARG1 in\n",
      "20900\n",
      "MYB in\n",
      "21000\n",
      "CITED2 in\n",
      "21100\n",
      "LATS1 in\n",
      "21200\n",
      "ESR1 in\n",
      "OPRM1 in\n",
      "21300\n",
      "SOD2 in\n",
      "SOD2 in\n",
      "WTAP in\n",
      "21400\n",
      "PLG in\n",
      "RNASET2 in\n",
      "21500\n",
      "DLL1 in\n",
      "21600\n",
      "21700\n",
      "PMS2 in\n",
      "RAC1 in\n",
      "21800\n",
      "TWIST1 in\n",
      "21900\n",
      "IL6 in\n",
      "SNX10 in\n",
      "22000\n",
      "HOXA10 in\n",
      "HOXA11 in\n",
      "22100\n",
      "AQP1 in\n",
      "22200\n",
      "SFRP4 in\n",
      "22300\n",
      "INHBA in\n",
      "GLI3 in\n",
      "22400\n",
      "IGFBP1 in\n",
      "IGFBP3 in\n",
      "22500\n",
      "EGFR in\n",
      "PSPH in\n",
      "22600\n",
      "22700\n",
      "22800\n",
      "22900\n",
      "FZD9 in\n",
      "CCL24 in\n",
      "POR in\n",
      "23000\n",
      "FGL2 in\n",
      "23100\n",
      "HGF in\n",
      "ABCB1 in\n",
      "23200\n",
      "CDK6 in\n",
      "HEPACAM2 in\n",
      "PON1 in\n",
      "23300\n",
      "MCM7 in\n",
      "23400\n",
      "EPO in\n",
      "ACHE in\n",
      "SERPINE1 in\n",
      "23500\n",
      "23600\n",
      "NRCAM in\n",
      "DOCK4 in\n",
      "TES in\n",
      "MET in\n",
      "23700\n",
      "WNT2 in\n",
      "CFTR in\n",
      "WNT16 in\n",
      "23800\n",
      "SMO in\n",
      "23900\n",
      "24000\n",
      "24100\n",
      "BRAF in\n",
      "24200\n",
      "24300\n",
      "NOBOX in\n",
      "EZH2 in\n",
      "24400\n",
      "GIMAP7 in\n",
      "GIMAP6 in\n",
      "24500\n",
      "SHH in\n",
      "24600\n",
      "CD99 in\n",
      "STS in\n",
      "24700\n",
      "HCCS in\n",
      "CA5B in\n",
      "24800\n",
      "24900\n",
      "NR0B1 in\n",
      "25000\n",
      "25100\n",
      "RGN in\n",
      "TIMP1 in\n",
      "25200\n",
      "PORCN in\n",
      "WAS in\n",
      "TFE3 in\n",
      "FOXP3 in\n",
      "25300\n",
      "BMP15 in\n",
      "25400\n",
      "25500\n",
      "MSN in\n",
      "AR in\n",
      "25600\n",
      "MED12 in\n",
      "CXCR3 in\n",
      "25700\n",
      "PGK1 in\n",
      "25800\n",
      "25900\n",
      "26000\n",
      "26100\n",
      "26200\n",
      "26300\n",
      "XIAP in\n",
      "26400\n",
      "26500\n",
      "26600\n",
      "26700\n",
      "FMR1 in\n",
      "26800\n",
      "26900\n",
      "L1CAM in\n",
      "27000\n",
      "ANGPT2 in\n",
      "27100\n",
      "DEFB1 in\n",
      "27200\n",
      "27300\n",
      "GATA4 in\n",
      "27400\n",
      "27500\n",
      "HR in\n",
      "LOXL2 in\n",
      "STC1 in\n",
      "27600\n",
      "BNIP3L in\n",
      "CLU in\n",
      "FZD3 in\n",
      "27700\n",
      "27800\n",
      "STAR in\n",
      "FGFR1 in\n",
      "SFRP1 in\n",
      "ANK1 in\n",
      "27900\n",
      "28000\n",
      "MCM4 in\n",
      "SNAI2 in\n",
      "28100\n",
      "PLAG1 in\n",
      "CA8 in\n",
      "CHD7 in\n",
      "28200\n",
      "ADHFE1 in\n",
      "MYBL1 in\n",
      "28300\n",
      "PRDM14 in\n",
      "EYA1 in\n",
      "MSC in\n",
      "TRPA1 in\n",
      "28400\n",
      "28500\n",
      "CA1 in\n",
      "CA3 in\n",
      "CA2 in\n",
      "28600\n",
      "CCNE2 in\n",
      "28700\n",
      "STK3 in\n",
      "28800\n",
      "FZD6 in\n",
      "CTHRC1 in\n",
      "28900\n",
      "RSPO2 in\n",
      "29000\n",
      "29100\n",
      "MYC in\n",
      "PVT1 in\n",
      "29200\n",
      "NDRG1 in\n",
      "29300\n",
      "ARC in\n",
      "PSCA in\n",
      "CYP11B2 in\n",
      "29400\n",
      "29500\n",
      "29600\n",
      "29700\n",
      "RPS6 in\n",
      "29800\n",
      "CDKN2A in\n",
      "29900\n",
      "DDX58 in\n",
      "30000\n",
      "CA9 in\n",
      "30100\n",
      "30200\n",
      "30300\n",
      "30400\n",
      "30500\n",
      "ALDH1A1 in\n",
      "30600\n",
      "GOLM1 in\n",
      "30700\n",
      "ROR2 in\n",
      "30800\n",
      "FBP1 in\n",
      "30900\n",
      "TGFBR1 in\n",
      "NR4A3 in\n",
      "31000\n",
      "KLF4 in\n",
      "31100\n",
      "31200\n",
      "TNC in\n",
      "TLR4 in\n",
      "31300\n",
      "NR5A1 in\n",
      "HSPA5 in\n",
      "31400\n",
      "CDK9 in\n",
      "LCN2 in\n",
      "SET in\n",
      "PHYHD1 in\n",
      "31500\n",
      "PTGES in\n",
      "SETX in\n",
      "TTF1 in\n",
      "TSC1 in\n",
      "31600\n",
      "31700\n",
      "NOTCH1 in\n",
      "NDOR1 in\n",
      "31800\n",
      "IFITM3 in\n",
      "HRAS in\n",
      "31900\n",
      "CD151 in\n",
      "MUC6 in\n",
      "MUC2 in\n",
      "MUC5AC in\n",
      "MUC5B in\n",
      "H19 in\n",
      "IGF2 in\n",
      "TH in\n",
      "32000\n",
      "32100\n",
      "32200\n",
      "ILK in\n",
      "32300\n",
      "ADM in\n",
      "DKK3 in\n",
      "PTH in\n",
      "32400\n",
      "SPON1 in\n",
      "LDHA in\n",
      "32500\n",
      "32600\n",
      "PAX6 in\n",
      "WT1 in\n",
      "CD59 in\n",
      "32700\n",
      "CAT in\n",
      "CD44 in\n",
      "32800\n",
      "CD82 in\n",
      "F2 in\n",
      "32900\n",
      "33000\n",
      "33100\n",
      "33200\n",
      "33300\n",
      "CD5 in\n",
      "33400\n",
      "BAD in\n",
      "33500\n",
      "MEN1 in\n",
      "NEAT1 in\n",
      "MALAT1 in\n",
      "33600\n",
      "PC in\n",
      "33700\n",
      "AIP in\n",
      "GSTP1 in\n",
      "LRP5 in\n",
      "GAL in\n",
      "33800\n",
      "CCND1 in\n",
      "FGF19 in\n",
      "33900\n",
      "34000\n",
      "WNT11 in\n",
      "34100\n",
      "PRSS23 in\n",
      "34200\n",
      "FZD4 in\n",
      "34300\n",
      "34400\n",
      "PGR in\n",
      "YAP1 in\n",
      "MMP7 in\n",
      "MMP1 in\n",
      "MMP12 in\n",
      "34500\n",
      "SLN in\n",
      "ATM in\n",
      "FDX1 in\n",
      "34600\n",
      "CRYAB in\n",
      "SDHD in\n",
      "PTS in\n",
      "NNMT in\n",
      "34700\n",
      "34800\n",
      "34900\n",
      "35000\n",
      "ETS1 in\n",
      "35100\n",
      "35200\n",
      "35300\n",
      "GATA3 in\n",
      "35400\n",
      "VIM in\n",
      "35500\n",
      "BMI1 in\n",
      "35600\n",
      "BAMBI in\n",
      "35700\n",
      "ZEB1 in\n",
      "35800\n",
      "35900\n",
      "RET in\n",
      "CXCL12 in\n",
      "36000\n",
      "MAPK8 in\n",
      "WDFY4 in\n",
      "36100\n",
      "DKK1 in\n",
      "CDK1 in\n",
      "36200\n",
      "SIRT1 in\n",
      "TET1 in\n",
      "SRGN in\n",
      "HK1 in\n",
      "36300\n",
      "NODAL in\n",
      "SPOCK2 in\n",
      "DDIT4 in\n",
      "ECD in\n",
      "36400\n",
      "36500\n",
      "36600\n",
      "BMPR1A in\n",
      "GLUD1 in\n",
      "PTEN in\n",
      "36700\n",
      "ACTA2 in\n",
      "FAS in\n",
      "TNKS2 in\n",
      "36800\n",
      "36900\n",
      "SFRP5 in\n",
      "GOT1 in\n",
      "ABCC2 in\n",
      "WNT8B in\n",
      "PAX2 in\n",
      "FGF8 in\n",
      "37000\n",
      "LDB1 in\n",
      "NFKB2 in\n",
      "37100\n",
      "PDCD4 in\n",
      "ACSL5 in\n",
      "37200\n",
      "EMX2 in\n",
      "37300\n",
      "FGFR2 in\n",
      "DHX32 in\n",
      "37400\n",
      "BNIP3 in\n",
      "37500\n",
      "CYP2E1 in\n",
      "WNK1 in\n",
      "WNT5B in\n",
      "FBXL14 in\n",
      "37600\n",
      "FOXM1 in\n",
      "VWF in\n",
      "LTBR in\n",
      "GAPDH in\n",
      "37700\n",
      "CD4 in\n",
      "CDCA3 in\n",
      "NANOG in\n",
      "37800\n",
      "37900\n",
      "LRP6 in\n",
      "CDKN1B in\n",
      "38000\n",
      "GPRC5A in\n",
      "38100\n",
      "KRAS in\n",
      "38200\n",
      "38300\n",
      "38400\n",
      "38500\n",
      "VDR in\n",
      "38600\n",
      "WNT10B in\n",
      "WNT1 in\n",
      "DHH in\n",
      "38700\n",
      "38800\n",
      "SP1 in\n",
      "HOTAIR in\n",
      "38900\n",
      "RDH5 in\n",
      "CD63 in\n",
      "CDK2 in\n",
      "ERBB3 in\n",
      "39000\n",
      "STAT6 in\n",
      "NDUFA4L2 in\n",
      "GLI1 in\n",
      "DDIT3 in\n",
      "CDK4 in\n",
      "39100\n",
      "39200\n",
      "HMGA2 in\n",
      "CAND1 in\n",
      "IFNG in\n",
      "MDM2 in\n",
      "CPM in\n",
      "39300\n",
      "LGR5 in\n",
      "39400\n",
      "NTS in\n",
      "39500\n",
      "DUSP6 in\n",
      "39600\n",
      "39700\n",
      "IGF1 in\n",
      "39800\n",
      "ISCU in\n",
      "39900\n",
      "40000\n",
      "PTPN11 in\n",
      "SDS in\n",
      "40100\n",
      "40200\n",
      "ORAI1 in\n",
      "40300\n",
      "UBC in\n",
      "40400\n",
      "FZD10 in\n",
      "40500\n",
      "POLE in\n",
      "40600\n",
      "LATS2 in\n",
      "40700\n",
      "FGF9 in\n",
      "40800\n",
      "CDX2 in\n",
      "FLT1 in\n",
      "HMGB1 in\n",
      "40900\n",
      "BRCA2 in\n",
      "SOHLH2 in\n",
      "SMAD9 in\n",
      "41000\n",
      "FOXO1 in\n",
      "TNFSF11 in\n",
      "41100\n",
      "RB1 in\n",
      "41200\n",
      "CAB39L in\n",
      "DLEU1 in\n",
      "41300\n",
      "41400\n",
      "41500\n",
      "41600\n",
      "41700\n",
      "41800\n",
      "41900\n",
      "MCF2L in\n",
      "42000\n",
      "42100\n",
      "APEX1 in\n",
      "ANG in\n",
      "42200\n",
      "42300\n",
      "MMP14 in\n",
      "PRMT5 in\n",
      "42400\n",
      "42500\n",
      "42600\n",
      "FOXA1 in\n",
      "42700\n",
      "42800\n",
      "BMP4 in\n",
      "42900\n",
      "DAAM1 in\n",
      "43000\n",
      "HIF1A in\n",
      "43100\n",
      "MAX in\n",
      "ARG2 in\n",
      "43200\n",
      "43300\n",
      "43400\n",
      "43500\n",
      "43600\n",
      "GSC in\n",
      "43700\n",
      "43800\n",
      "43900\n",
      "44000\n",
      "AKT1 in\n",
      "44100\n",
      "44200\n",
      "44300\n",
      "44400\n",
      "SNHG14 in\n",
      "44500\n",
      "44600\n",
      "44700\n",
      "GREM1 in\n",
      "44800\n",
      "44900\n",
      "RAD51 in\n",
      "DLL4 in\n",
      "45000\n",
      "45100\n",
      "SHC4 in\n",
      "45200\n",
      "RAB27A in\n",
      "45300\n",
      "ADAM10 in\n",
      "CCNB2 in\n",
      "45400\n",
      "ANXA2 in\n",
      "CA12 in\n",
      "45500\n",
      "ANKDD1A in\n",
      "45600\n",
      "45700\n",
      "CYP11A1 in\n",
      "45800\n",
      "CSPG4 in\n",
      "45900\n",
      "46000\n",
      "46100\n",
      "MFGE8 in\n",
      "46200\n",
      "IDH2 in\n",
      "PRC1 in\n",
      "46300\n",
      "IGF1R in\n",
      "46400\n",
      "46500\n",
      "AXIN1 in\n",
      "46600\n",
      "TSC2 in\n",
      "PKD1 in\n",
      "PGP in\n",
      "46700\n",
      "46800\n",
      "46900\n",
      "47000\n",
      "47100\n",
      "47200\n",
      "GP2 in\n",
      "47300\n",
      "PLK1 in\n",
      "CHP2 in\n",
      "47400\n",
      "47500\n",
      "MVP in\n",
      "ALDOA in\n",
      "47600\n",
      "47700\n",
      "47800\n",
      "47900\n",
      "48000\n",
      "NKD1 in\n",
      "SALL1 in\n",
      "48100\n",
      "MMP2 in\n",
      "MT1G in\n",
      "CCL22 in\n",
      "48200\n",
      "KIFC3 in\n",
      "48300\n",
      "CA7 in\n",
      "CTCF in\n",
      "48400\n",
      "NFATC3 in\n",
      "CDH3 in\n",
      "CDH1 in\n",
      "CYB5B in\n",
      "NFAT5 in\n",
      "NQO1 in\n",
      "48500\n",
      "TAT in\n",
      "HP in\n",
      "48600\n",
      "48700\n",
      "ADAMTS18 in\n",
      "48800\n",
      "48900\n",
      "FOXC2 in\n",
      "MVD in\n",
      "49000\n",
      "49100\n",
      "TRPV1 in\n",
      "49200\n",
      "CXCL16 in\n",
      "49300\n",
      "DVL2 in\n",
      "49400\n",
      "CD68 in\n",
      "TP53 in\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "49500\n",
      "49600\n",
      "TRIM16 in\n",
      "49700\n",
      "SHMT1 in\n",
      "49800\n",
      "49900\n",
      "50000\n",
      "FOXN1 in\n",
      "50100\n",
      "CPD in\n",
      "50200\n",
      "EVI2B in\n",
      "50300\n",
      "50400\n",
      "TBC1D3 in\n",
      "50500\n",
      "ERBB2 in\n",
      "TOP2A in\n",
      "50600\n",
      "50700\n",
      "STAT5A in\n",
      "STAT3 in\n",
      "BRCA1 in\n",
      "50800\n",
      "FZD2 in\n",
      "GFAP in\n",
      "50900\n",
      "WNT3 in\n",
      "WNT9B in\n",
      "51000\n",
      "NGFR in\n",
      "51100\n",
      "COL1A1 in\n",
      "TOB1 in\n",
      "CA10 in\n",
      "51200\n",
      "MMD in\n",
      "NOG in\n",
      "COIL in\n",
      "51300\n",
      "CA4 in\n",
      "51400\n",
      "ACE in\n",
      "PECAM1 in\n",
      "51500\n",
      "AXIN2 in\n",
      "PRKCA in\n",
      "51600\n",
      "PRKAR1A in\n",
      "SOX9 in\n",
      "51700\n",
      "FDXR in\n",
      "51800\n",
      "51900\n",
      "CBX2 in\n",
      "CBX4 in\n",
      "52000\n",
      "FASN in\n",
      "SLC16A3 in\n",
      "52100\n",
      "52200\n",
      "52300\n",
      "52400\n",
      "52500\n",
      "IMPACT in\n",
      "52600\n",
      "CDH2 in\n",
      "DSG2 in\n",
      "TTR in\n",
      "52700\n",
      "52800\n",
      "SMAD2 in\n",
      "52900\n",
      "SMAD4 in\n",
      "DCC in\n",
      "TCF4 in\n",
      "53000\n",
      "GRP in\n",
      "PMAIP1 in\n",
      "53100\n",
      "BCL2 in\n",
      "53200\n",
      "53300\n",
      "NFATC1 in\n",
      "CSNK2A1 in\n",
      "53400\n",
      "PRNP in\n",
      "53500\n",
      "PCNA in\n",
      "BMP2 in\n",
      "PLCB1 in\n",
      "JAG1 in\n",
      "53600\n",
      "53700\n",
      "INSM1 in\n",
      "FOXA2 in\n",
      "53800\n",
      "53900\n",
      "ID1 in\n",
      "BCL2L1 in\n",
      "54000\n",
      "E2F1 in\n",
      "GDF5 in\n",
      "54100\n",
      "NFS1 in\n",
      "TGM2 in\n",
      "54200\n",
      "54300\n",
      "STK4 in\n",
      "MMP9 in\n",
      "CD40 in\n",
      "54400\n",
      "SNAI1 in\n",
      "54500\n",
      "BMP7 in\n",
      "54600\n",
      "GNAS in\n",
      "54700\n",
      "54800\n",
      "PALM in\n",
      "54900\n",
      "TCF3 in\n",
      "AMH in\n",
      "55000\n",
      "SIRT6 in\n",
      "EBI3 in\n",
      "55100\n",
      "C3 in\n",
      "55200\n",
      "ANGPTL4 in\n",
      "55300\n",
      "55400\n",
      "SPC24 in\n",
      "EPOR in\n",
      "55500\n",
      "BEST2 in\n",
      "JUNB in\n",
      "55600\n",
      "NOTCH3 in\n",
      "BRD4 in\n",
      "55700\n",
      "KLF2 in\n",
      "PGLS in\n",
      "55800\n",
      "JUND in\n",
      "GDF15 in\n",
      "COPE in\n",
      "55900\n",
      "56000\n",
      "56100\n",
      "CCNE1 in\n",
      "56200\n",
      "HPN in\n",
      "56300\n",
      "56400\n",
      "MAP4K1 in\n",
      "56500\n",
      "DLL3 in\n",
      "FCGBP in\n",
      "AKT2 in\n",
      "56600\n",
      "MIA in\n",
      "AXL in\n",
      "TGFB1 in\n",
      "PAFAH1B3 in\n",
      "56700\n",
      "CEACAM1 in\n",
      "56800\n",
      "PVR in\n",
      "APOE in\n",
      "ERCC1 in\n",
      "56900\n",
      "HIF3A in\n",
      "PRKD2 in\n",
      "BBC3 in\n",
      "57000\n",
      "BAX in\n",
      "57100\n",
      "AKT1S1 in\n",
      "KLK6 in\n",
      "KLK7 in\n",
      "57200\n",
      "57300\n",
      "57400\n",
      "57500\n",
      "IL11 in\n",
      "57600\n",
      "57700\n",
      "57800\n",
      "SRY in\n",
      "57900\n",
      "58000\n",
      "58100\n",
      "58200\n",
      "58300\n",
      "58400\n",
      "BID in\n",
      "58500\n",
      "58600\n",
      "MAPK1 in\n",
      "58700\n",
      "58800\n",
      "SMARCB1 in\n",
      "MIF in\n",
      "58900\n",
      "XBP1 in\n",
      "59000\n",
      "EWSR1 in\n",
      "LIF in\n",
      "59100\n",
      "C22orf42 in\n",
      "TIMP3 in\n",
      "59200\n",
      "MB in\n",
      "APOL2 in\n",
      "59300\n",
      "SOX10 in\n",
      "BAIAP2L2 in\n",
      "PDGFB in\n",
      "ATF4 in\n",
      "59400\n",
      "59500\n",
      "PRR5 in\n",
      "WNT7B in\n",
      "59600\n",
      "59700\n",
      "59800\n",
      "59900\n",
      "60000\n",
      "60100\n",
      "SOD1 in\n",
      "60200\n",
      "RUNX1 in\n",
      "60300\n",
      "ERG in\n",
      "MX1 in\n",
      "60400\n",
      "TFF3 in\n",
      "TFF1 in\n",
      "60500\n",
      "COL18A1 in\n",
      "COL6A1 in\n",
      "60600\n",
      "MT-CO1 in\n"
     ]
    }
   ],
   "source": [
    "data_gene_names = []\n",
    "with open(file_gene_path) as f:\n",
    "    for line in f.readlines():\n",
    "        data_gene_names.append(line.strip())\n",
    "\n",
    "gene_sel = []\n",
    "for idx, gene_name in enumerate(data_gene_names):\n",
    "    if idx % 100 == 0:\n",
    "        print(idx)\n",
    "    gene_name_regex = '([ ]|^)' + gene_name.replace('.', '\\.').replace('-', '\\-') + '([ ,]|$)'\n",
    "    if re.search(gene_name_regex, word_str, re.I):\n",
    "        print(gene_name, 'in')\n",
    "        gene_sel.append(gene_name)\n",
    "gene_sel.append('Carbonic anhydrase IX')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "7a632656",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    " 'NDRG1' in gene_sel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "1aac0b9c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "768\n",
      "767\n"
     ]
    }
   ],
   "source": [
    "print(len(gene_sel))\n",
    "gene_sel = list(set(gene_sel))\n",
    "print(len(gene_sel))\n",
    "gene_sel=sorted(gene_sel, reverse=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "25649a4c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['WWOX-AS1', 'WWOX', 'VSTM2L', 'VDAC1P9', 'VAT1L', 'TTI1', 'TLDC2', 'TGM2', 'TGIF2-RAB5IF', 'TGIF2', 'SRC', 'SPTA1', 'SOGA1', 'SNORA71E', 'SNORA71D', 'SNORA71C', 'SNORA71B', 'SNORA71A', 'SNORA60', 'SNHG17', 'SNHG11', 'SMU1P1', 'SLC32A1', 'SLA2', 'SH2D2A', 'SDR42E1', 'SCAND1', 'SAMHD1', 'RPS3P7', 'RPS3P2', 'RPS3AP3', 'RPS27AP3', 'RPRD1B', 'RPN2', 'RPL7AP14', 'RPF2P1', 'ROMO1', 'RNU6-937P', 'RNU6-1191P', 'RNU4-40P', 'RNA5SP60', 'RNA5SP431', 'RN7SL612P', 'RN7SL237P', 'RN7SL156P', 'RN7SL116P', 'RN7SKP185', 'RN7SKP176', 'RN7SKP173', 'RBM39', 'RBL1', 'RALGAPB', 'RAD1P2', 'RAB5IF', 'PYHIN5P', 'PYHIN1', 'PRCC', 'PPP1R16B', 'PPIAP51', 'PPIAP50', 'PPIAP3', 'PLCG2', 'PKD1L2', 'PHF20', 'PEAR1', 'OR6Y1', 'OR6P1', 'OR6N2', 'OR6N1', 'OR6K6', 'OR6K5P', 'OR6K4P', 'OR6K3', 'OR6K2', 'OR6K1P', 'OR2AQ1P', 'OR10Z1', 'OR10X1', 'OR10T2', 'OR10T1P', 'OR10R3P', 'OR10R2', 'OR10R1P', 'OR10K2', 'OR10K1', 'OR10J9P', 'OR10J8P', 'OR10J7P', 'OR10J6P', 'OR10J5', 'OR10J4', 'OR10J3', 'OR10J2P', 'OR10J1', 'OR10AE1P', 'OR10AA1P', 'NUDT7', 'NTRK1', 'NORAD', 'NNAT', 'NFS1', 'NDRG3', 'MYL9', 'MTCYBP28', 'MRPS21P2', 'MROH8', 'MPTX1', 'MPHOSPH6', 'MNDA', 'MIR7854', 'MIR765', 'MIR6504', 'MIR548O2', 'MIR4720', 'MANBAL', 'MAFTRR', 'MAF', 'LSM3P5', 'LRRC71', 'LINC02819', 'LINC02772', 'LINC02131', 'LINC01746', 'LINC01704', 'LINC01229', 'LINC01228', 'LINC01227', 'LINC00489', 'LBP', 'KRT8P45', 'KRT8P22', 'KIRREL1-IT1', 'KIRREL1', 'KIAA1755', 'INSRR', 'IFI16', 'HSD17B2', 'HNRNPA3P2', 'HMGN1P5', 'HMGB3P2', 'HIGD1AP16', 'GLRXP1', 'GHRH', 'GCSH', 'GAN', 'FCRL5', 'FCRL4', 'FCRL3', 'FCRL2', 'FCRL1', 'FCER1A', 'FAM83D', 'ETV3L', 'ETV3', 'EPB41L1', 'ELL2P1', 'EI24P2', 'DYNLRB2', 'DSN1', 'DLGAP4-AS1', 'DLGAP4', 'CYCSP52', 'CTNNBL1', 'COX7BP2', 'CNBD2', 'CMIP', 'CMC2', 'CLEC3A', 'CENPN', 'CDYL2', 'CD5L', 'CD1E', 'CD1D', 'CD1C', 'CD1B', 'CD1A', 'CADM3-AS1', 'CADM3', 'C16orf46', 'BPI', 'BLCAP', 'BCO1', 'ATMIN', 'ARLNC1', 'ARHGEF11', 'ARHGAP40', 'APCS', 'AL732406.1', 'AL606752.1', 'AL590666.5', 'AL513323.1', 'AL445528.2', 'AL391095.5', 'AL391095.4', 'AL391095.3', 'AL391095.2', 'AL391095.1', 'AL365440.2', 'AL365440.1', 'AL359555.4', 'AL359555.3', 'AL359555.2', 'AL359555.1', 'AL357374.1', 'AL357143.2', 'AL357143.1', 'AL356276.2', 'AL356276.1', 'AL353899.1', 'AL162293.1', 'AL139010.1', 'AL138900.2', 'AL138900.1', 'AL138899.3', 'AL138899.2', 'AL138899.1', 'AL136172.1', 'AL135929.2', 'AL135929.1', 'AL133519.1', 'AL121895.2', 'AL121895.1', 'AL109614.1', 'AL078461.1', 'AL050318.1', 'AL035420.3', 'AL035420.2', 'AL035420.1', 'AL035419.1', 'AL034422.1', 'AL031662.1', 'AL031659.1', 'AL031651.2', 'AL031651.1', 'AIM2', 'ADIG', 'ADAMTS18', 'ACTR5', 'ACKR1', 'AC136603.1', 'AC131888.1', 'AC108097.1', 'AC106743.1', 'AC105411.1', 'AC099524.1', 'AC099313.1', 'AC092724.1', 'AC092718.9', 'AC092718.8', 'AC092718.7', 'AC092718.6', 'AC092718.5', 'AC092718.4', 'AC092718.3', 'AC092718.2', 'AC092718.1', 'AC092376.3', 'AC092376.2', 'AC092376.1', 'AC092332.1', 'AC092142.2', 'AC092142.1', 'AC092139.3', 'AC092139.2', 'AC092139.1', 'AC092135.3', 'AC092135.2', 'AC092135.1', 'AC092134.3', 'AC092134.2', 'AC092134.1', 'AC092130.1', 'AC084064.1', 'AC079414.3', 'AC079414.2', 'AC079414.1', 'AC046158.4', 'AC046158.3', 'AC046158.2', 'AC046158.1', 'AC027279.4', 'AC027279.3', 'AC027279.2', 'AC027279.1', 'AC025284.1', 'AC022166.2', 'AC022166.1', 'AC009159.4', 'AC009159.3', 'AC009159.2', 'AC009159.1', 'AC009148.1', 'AC009145.4', 'AC009145.3', 'AC009145.2', 'AC009145.1', 'AC009141.1', 'AC009070.1', 'AC009044.1', 'AAR2']\n"
     ]
    }
   ],
   "source": [
    "print(gene_sel)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "e53f6a99",
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_sel.append('Carbonic anhydrase IX')\n",
    "gene_sel.append('carbonic anhydrase IX')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "01b4c642",
   "metadata": {},
   "source": [
    "### 拆分训练测试集"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "37509f92",
   "metadata": {},
   "outputs": [],
   "source": [
    "import random\n",
    "random.shuffle(data_cls)\n",
    "train_len = int(len(data_cls) * 0.8)\n",
    "data_train, data_test = data_cls[:train_len], data_cls[train_len:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "891aa34c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(1650, 413)"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(data_train), len(data_test)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5d59be51",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "d6edf0a7",
   "metadata": {},
   "source": [
    "### 训练集分析"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1c649a5e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "gene_sel_trans = [gene.replace('.', '\\.').replace('-', '\\-') for gene in gene_sel]\n",
    "gene_regex = '((([ ]|^)' + '([ .,]|$))|(([ ]|^)'.join(gene_sel_trans) + '([ .,]|$)))'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "22f77ead",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Compared to normal adult colon, all mouse and human tumors over-expressed a large cluster of genes highly enriched for functional association to the control of cell cycle progression, proliferation, and migration, including those encoding MYC, AKT2, PLK1 and SPARC.', ['AKT2', 'SPARC', 'PLK1', 'MYC'])\n",
      "(' Human CRC adenocarcinomas lost an additional suppressor module (IGFBP4, MAP4K1, PDGFRA, STAB1 and WNT4).', ['PDGFRA', 'STAB1', 'MAP4K1'])\n",
      "(' Many human tumor samples also gained expression of a coordinately regulated module associated with advanced malignancy (ABCC1, FOXO3A, LIF, PIK3R1, PRNP, TNC, TIMP3 and VEGF).', ['TIMP3', 'PRNP', 'LIF', 'PIK3R1', 'TNC'])\n",
      "---------------------\n",
      "(' Genes overexpressed in breast carcinoma included TFF1, TFF3, FOXA1, CA12, GATA3, SDC1, PITX1, TH, EHFD1, EFEMP1, TOB1 and KLF2.', ['TFF3', 'KLF2', 'TH', 'PITX1', 'TOB1', 'EFEMP1', 'TFF1', 'CA12', 'GATA3', 'SDC1', 'FOXA1'])\n",
      "(' Genes overexpressed in ovarian/peritoneal carcinoma included SPON1, RBP1, MFGE8, TM4SF12, MMP7, KLK5/6/7, FOLR1/3,PAX8, APOL2 and NRCAM.', ['APOL2', 'SPON1', 'RBP1', 'MFGE8', 'MMP7', 'NRCAM'])\n",
      "---------------------\n",
      "(' Several members of the WNT pathway, including WNT5A, FZD5, and DKK1 were highly up-regulated in PCa tissue from patients with advanced PCa.', ['FZD5', 'WNT5A', 'DKK1'])\n",
      "(' While WNT5A, FZD5, and DKK1 were highly expressed in PC3 cells, WNT1 and SFRP1 mRNA levels were higher in DU145 cells.', ['FZD5', 'SFRP1', 'WNT5A', 'DKK1', 'WNT1'])\n",
      "(' Moreover, zoledronic acid down-regulated mRNA levels of WNT5A (-34%), FZD5 (-60%), and DKK1 (-46%) in PC3 cells.', ['FZD5', 'WNT5A', 'DKK1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In addition, fenofibrate-enhanced HO-1 upregulation activates SIRT1 and leads to subsequent accumulation of SIRT1 in the nucleus, which further promotes HIF-1α deacetylation and inhibits CA9 expression.', ['CA9', 'SIRT1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemical staining of tumor tissue sections confirmed the increased expression of CAIX, HK2 and LDHA in ccRCC, validating the microarray data and supporting their potential as the energetic metabolism-related biomarkers of the ccRCC.', ['LDHA', 'HK2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The present study additionally identified 3 potential genes (lysyl oxidase, ankyrin repeat domain 37, B-cell lymphoma 2 interacting protein 3 like) previously identified in other studies, which may be considered as universal hypoxia genes along with HIF1α, EPO, VEGF, glucose transporter 1 (GLUT1), CA9, and LDH.', ['CA9', 'EPO'])\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: ZEB1 bound to the promoter of CA9 to positively regulate CA9 expression in tongue cancer cells.', ['CA9', 'ZEB1'])\n",
      "(' Knockdown of CA9 using short interfering RNA (siRNA) abolished the chemoresistance resulting from ZEB1 overexpression in Tca8113 and SCC-25 cells, and CA9 overexpression attenuated chemosensitivity induced by ZEB1 knockdown in Tca8113/PYM cells.', ['CA9', 'ZEB1'])\n",
      "(' CA9 knockdown also prevented maintenance of pHi mediated by overexpression of ZEB1 in Tca8113 and SCC-25 cells following chemotherapy, associated with increased apoptosis and caspase-3 activation.', ['CA9', 'ZEB1'])\n",
      "(' Conversely, ectopic expression of CA9 suppressed decrease in pHi mediated by ZEB1 knockdown in Tca8113/PYM cells following chemotherapy, accompanied by decreased apoptosis and caspase-3 activation.', ['CA9', 'ZEB1'])\n",
      "(' Importantly, a positive correlation was observed between ZEB1 and CA9 protein expression in tongue cancer tissues, and expression of these proteins associated with a poor prognosis for patients.', ['CA9', 'ZEB1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In this study, human MMNK-1 cholangiocytes with repressed CFTR function were used to induce SULT1E1 expression in human HepG2 hepatocytes to investigate whether SULT1E1 can be increased in human CF liver.', ['SULT1E1', 'CFTR'])\n",
      "(' Results demonstrate that inhibition of CFTR expression in MMNK-1 cells results in the induction of SULT1E1 message and activity in HepG2 cells in the Transwell system.', ['SULT1E1', 'CFTR'])\n",
      "(' Thus, we have shown that co-culture of HepG2 hepatocytes with MMNK-1 cholangiocytes with siRNA repressed CFTR expression results in the selective induction of SULT1E1 in the HepG2 cells.', ['SULT1E1', 'CFTR'])\n",
      "(' Loss of CFTR function in cholangiocytes may have a paracrine regulatory effect on hepatocytes via the induction of SULT1E1 and the increased sulfation of beta-estradiol.', ['SULT1E1', 'CFTR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The TST15 gene panel and HRAS sequencing revealed no pathogenic mutations in BRAF, KRAS, EGFR, AKT1, or HRAS.', ['EGFR', 'AKT1', 'HRAS', 'KRAS', 'BRAF'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Therefore, we analyzed the CA9 promoter activity in transfected tumor cells expressing RET and/or the HIF-α subunit.', ['CA9', 'RET'])\n",
      "(' We showed that overexpression of both wild-type and mutant RET can increase the CA9 promoter activity induced by HIF-1 (but not HIF-2) in hypoxia.', ['CA9', 'RET'])\n",
      "(' Moreover, inhibition of the major kinase pathways, which transmit signals from RET and regulate HIF-1, abrogated their cooperative effect on the CA9 promoter.', ['CA9', 'RET'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(\"RESULTS: Cancer and Alzheimer's disease have inverse relationship in many aspects such as P53, estrogen, neurotrophins and growth factors, growth and proliferation, cAMP, EGFR, Bcl-2, apoptosis pathways, IGF-1, HSV, TDP-43, APOE variants, notch signals and presenilins, NCAM, TNF alpha, PI3K/AKT/MTOR pathway, telomerase, ROS, ACE levels.\", ['EGFR', 'ACE', 'TNF', 'APOE'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' CA4, PECAM1, DNAJB4, AGER, GIMAP6, C10orf54 and DOCK4 were expressed at lower levels in the tumor samples, whereas GOLM1 and PAFAH1B3 were highly expressed in tumor samples.', ['DOCK4', 'DNAJB4', 'AGER', 'GIMAP6', 'PECAM1', 'GOLM1', 'CA4', 'PAFAH1B3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' RT-qPCR was used to measure mRNA levels of WNT family members (WNT1, WNT3, WNT4, WNT5, sFRP4 and sFRP5).', ['WNT3', 'WNT4'])\n",
      "(' The PCOS group had higher transcript levels of WNT1, WNT3 and WNT4 (p<0.', ['WNT1', 'WNT3', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Among them, six up-regulated proteins (carbonic anhydrase 3 (Ca 3), Hsp-27, S100A6, galectin-7, S100A9, S100A11) and one down-regulated protein (superoxide dismutase [Cu-Zn] (Sod 1)) are associated with cancer-related key processes.', ['S100A6', 'S100A9'])\n",
      "(' Comparative studies showed that Ca 3, S100A6, S100A9, S100A11 and Sod 1 are specific for stages of development and progression of tumors whereas Hsp-27 and galectin-7 are specific for tumor promotion stage by cypermethrin in mouse skin.', ['S100A11', 'S100A6', 'S100A9'])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "(' The purpose of this study was to investigate mutations in 50 oncogenes and tumor suppressor genes, including APC and CTNNB1, in which mutations in COF have been previously reported.', ['APC', 'CTNNB1'])\n",
      "('RESULTS: The expression assay revealed 12 differentially expressed Wnt/β-catenin pathway genes in COF, including the upregulation of CTNNB1, TCF7, NKD1, and WNT5 A, and downregulation of CTNNBIP1, FRZB, FZD6, RHOU, SFRP4, WNT10 A, WNT3 A, and WNT4, suggesting activation of the Wnt/β-catenin signaling pathway.', ['WNT4', 'NKD1', 'RHOU', 'CTNNBIP1', 'CTNNB1', 'FZD6', 'WNT3', 'FRZB', 'TCF7', 'SFRP4'])\n",
      "(' NGS revealed 5 single nucleotide variants: TP53 (rs1042522), PIK3 CA (rs2230461), MET (rs33917957), KIT (rs3822214), and APC (rs33974176), but none of them was pathogenic.', ['TP53', 'MET', 'APC', 'KIT'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Leptin signaling increased the expression of tumorigenesis and chemoresistance-related genes (ABCB1, WNT4, ADHFE1, TBC1D3, LL22NC03, RDH5, and ITGB3) and impaired chemotherapeutic effects in TNBC cells.', ['ADHFE1', 'WNT4', 'TBC1D3', 'RDH5'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The downregulation of ETS1 expression with small interfering RNA (siRNA) inhibited the upregulation of CA9 and NDRG1/Cap43 caused by increased cell density.', ['CA9', 'ETS1'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Among familial cases, mutations of the succinate dehydrogenase (SDH) subunit genes (succinate dehydrogenase subunit [SDH]B, SDHC, SDHD, succinate dehydrogenase subunit AF2 [SDHAF2] , and SDHA) are the most common cause.', ['SDHD', 'SDHC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Under equivalent dosage of Dox, free Dox substantially changed the expression of ANK1, ACTA2, GPR87, GDF15, FZD6, and WNT4 in MCF-7 cells.', ['WNT4', 'ACTA2', 'ANK1', 'FZD6', 'GDF15', 'GPR87'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' BRCA1 inhibits NEAT1 expression potentially through binding to its genomic binding site upstream of the NEAT1 gene.', ['BRCA1', 'NEAT1'])\n",
      "(' BRCA1 deficiency in human normal/cancerous breast cells and mouse mammary glands leads to NEAT1 overexpression.', ['BRCA1', 'NEAT1'])\n",
      "(' Our studies show that NEAT1 upregulation resulting from BRCA1 deficiency stimulates in vitro and in vivo breast tumorigenicity.', ['BRCA1', 'NEAT1'])\n",
      "(' Silencing of miR-129-5p expression by NEAT1 results in upregulation of WNT4 expression, a target of miR-129-5p, which leads to activation of oncogenic WNT signaling.', ['WNT4', 'NEAT1'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' We examined 4 OS cell lines for WNT1, WNT4, WNT5A, WNT7A, WNT11, FZD1-10 and LRP5 expression by reverse transcription polymerase chain reaction (RT-PCR).', ['WNT4', 'WNT11', 'WNT7A', 'WNT5A', 'WNT1', 'LRP5'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Xuan-Fu-Hua decoction treatment significantly increased the transcription levels of DDIT3, PMAIP1, and ZMAT3 genes while decreasing that of WNT4, AXIN2, NFE2L2, TGFBR1, MITF, and IGFBP3 genes.', ['ZMAT3', 'WNT4', 'NFE2L2', 'DDIT3', 'AXIN2', 'IGFBP3', 'TGFBR1', 'MITF', 'PMAIP1'])\n",
      "---------------------\n",
      "('2, CD10, c-erbB-2, EGFR, HIF-1a, Ki67, MDM2, PAX-2 y 8, p53, survivin and VEGFR 1 and 2).', ['EGFR', 'MDM2'])\n",
      "(' Deletion-duplication of genes increased the likelihood: of death (APC, Bcl-2 and CDKN2A by 11, 7 and 4 respectively and SMAD4 reduced the probability by 88%); tumor recurrence (CDKN2A by fifteen fold and VHL reduced the probability by 87%); pT greater than 2 (CCND2, MDM2 and WT1 multiplied by 6, 7 and 9); risk of N+ (CDK4 and EBF1 by 13); distant metastases (BRCA2 and DLEU1 by 5); Fuhrman grade ≥3 (BRCA1, BRCA2 and p53 by 40, 75 and 34 respectively, while that FHIT reduced by 96%).', ['WT1', 'VHL', 'FHIT', 'DLEU1', 'EBF1', 'SMAD4', 'BRCA2', 'MDM2', 'CDKN2A'])\n",
      "(' Deletion-duplication of CDK4 and DCC increased survival by a factor of 13 and 16, while that DLEU1 and RUNX1 decreased survival time by 80%.', ['DCC', 'CDK4', 'DLEU1', 'RUNX1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Experimental studies have suggested that herbal compounds regulate the expression of many lncRNAs involved in cancer, such as HOTAIR (HOX transcript antisense RNA), H19, MALAT1 (metastasis-associated lung adenocarcinoma transcript 1), PCGEM1 (Prostate cancer gene expression marker 1), PVT1, etc.', ['PCGEM1', 'MALAT1', 'PVT1', 'H19', 'HOTAIR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Furthermore, ectopic expression of DVL1, LRP8 and Notch2 in malignant lung tissue validates the biological impact of these genetic alterations.', ['DVL1', 'LRP8'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The effects of sustained production of WNT4 were concomitant with upregulation of proliferative markers (c-Myc, Cyclin D1), the survival marker BCL-XL, the putative WNT4 receptor FZD6 and activation of ERK1 and STAT3.', ['FZD6', 'WNT4', 'STAT3'])\n",
      "(' In xenograft studies, tumors with autocrine hGH expressed higher levels of WNT4 and FZD6 when compared with control tumors.', ['FZD6', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('Silencing PTEN in the fallopian tube promotes enrichment of cancer stem cell-like function through loss of PAX2.', ['PAX2', 'PTEN'])\n",
      "(' In the present study, loss of PTEN in FTE led to the enrichment of cancer stem cell markers such as LGR5, WNT4, ALDH1, CD44.', ['CD44', 'WNT4', 'PTEN', 'LGR5'])\n",
      "(' Taken together, these data suggest that loss of PTEN induces reprogramming of the FTE cells into a more stem-like phenotype due to loss of PAX2 and provides a model to study early events during the FTE-driven ovarian cancer tumor formation.', ['PAX2', 'PTEN'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' PBMC expressions of six genes from our previous study (ANXA3, ARG1, CA5B, F5, SSBP2, and TBC1D8) along with four new genes (MIC1, NGAL, MUC1, and MUC16) were analyzed using multiplex Q-RT PCR.', ['SSBP2', 'ARG1', 'MUC1', 'F5', 'CA5B'])\n",
      "(\" We found that addition of PBMC CA5B, F5, SSBP2, and MIC1 expression levels to CA19-9 significantly improved CA19-9's diagnostic abilities when comparing resectable PC to CP patients (p=0.\", ['PC', 'SSBP2', 'CP', 'F5', 'CA5B'])\n",
      "(' We identified a score-based model that can differentiate resectable PC from CP better than CA19-9, potentiating that PBMC differential expression analysis may offer a novel tool for early PC diagnosis.', ['PC', 'CP'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We performed a precise characterization of mesenchymal differentiation using PAX2, WNT4 and WT1 as a marker, which revealed that impairment of the differentiation process preceded the development of cysts in Pkd1(-/-) mice.', ['WT1', 'WNT4', 'PAX2'])\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "(' Low levels of WNT2, WNT4, WNT5A have also been observed.', ['WNT4', 'WNT2', 'WNT5A'])\n",
      "(' Decreased expression of WNT2 and WNT5A may be caused by miR-331-3p and miR-200b-5p, respectively.', ['WNT2', 'WNT5A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemistry showed that atypical cells were positive for cytokeratin AE1&3, CK7, CD10, GATA3, glypican 3, Hep Par 1, carbonic anhydrase 9 (focal), and vimentin (focal) but negative for CK20, CD56, chromogranin A, synaptophysin, TTF1, HMB45, melan A, and PSA.', ['GATA3', 'TTF1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' A direct association of CA9 expression with epidermal growth factor receptor, c-erbB-2, and MUC1 expression was also noted (P < 0.', ['CA9', 'MUC1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Tumors were negative for cytokeratin 20, carbonic anhydrase IX, parvalbumin, HMB45, TTF1, TFE3, chromogranin A, thyroglobulin, PAX8, and ALK.', ['ALK', 'PAX8', 'TFE3', 'TTF1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('PADI4 stimulates esophageal squamous cell carcinoma tumor growth and up-regulates CA9 expression.', ['CA9', 'PADI4'])\n",
      "(' Immunohistochemistry detected the increased expression and co-localization of PADI4 and CA9 in ESCC tissues compared with adjacent non-tumor tissues and normal tissue controls.', ['CA9', 'PADI4'])\n",
      "(' The above findings suggest that increased PADI4 expression in ESCC stimulates tumor growth and up-regulates CA9 expression, which is known to promote metastatic properties in tumor cells.', ['CA9', 'PADI4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Finally, COX-2 and PTGES were colocalized in primary tumors along with HIF-1alpha and IGFBP3.', ['PTGES', 'IGFBP3'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' It was also demonstrated that MB was upregulated in hypoxic microregions of glioblastoma tumor xenografts that were stained in matched tumor regions of serial tumor sections with the hypoxia markers, pimonidazole, CAIX, VEGF and LDHA.', ['LDHA', 'MB'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The low MVD, increased HIF-1alpha protein levels, dissociation of hypoxia, and TP53 protein induction in the metastatic tumor cells (AN3CA) support a role for hypoxia in the development of recurrent endometrial carcinoma.', ['TP53', 'MVD'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Previously we showed that LOX is strongly induced upon hypoxia in nasopharyngeal carcinoma (NPC) cell lines CNE2 and HONE1 but only slightly in HK1 and not in C666-1.', ['HK1', 'LOX'])\n",
      "(' LOX was expressed in most NPC cell lines except for C666-1, while HK1 and FaDu (laryngeal cancer) only expressed low level of LOX.', ['HK1', 'LOX'])\n",
      "(' Methylation analysis showed that the LOX promoter was methylated in C666-1 and partially methylated in HK1.', ['HK1', 'LOX'])\n",
      "(' Carbonic anhydrase IX (CA9) mRNA level was obviously decreased in HK1 cells after transfection with LOX.', ['HK1', 'Carbonic anhydrase IX', 'LOX'])\n",
      "(' The elevation of CA9 protein upon hypoxia was inhibited in LOX-transfected HK1 cells.', ['CA9', 'HK1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(\" CA4 interacted with Wilms' tumour 1-associating protein (WTAP) and induced WTAP protein degradation through polyubiquitination.\", ['CA4', 'WTAP'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We tested the contribution of four SF-inducible/c-Akt-dependent genes (AMPD3, EPHB2, MX1 and WNT4) to protection against adriamycin (a DNA topoisomerase IIalpha inhibitor) using RNA interference.', ['MX1', 'EPHB2'])\n",
      "---------------------\n",
      "(' These positive effects of SIRT6 were associated with downregulation of the circulatory chemokine, CXCL10, and the myokine, WNT4.', ['WNT4', 'SIRT6', 'CXCL10'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Furthermore, HIF-1α, CA9, CD34, VEGFR1 and -3 and PDGRFα showed significant associations with progression-free survival (PFS) and overall survival (OS).', ['CA9', 'CD34'])\n",
      "(' Frequently observed copy number loss and mutation of VHL gene lead to altered expression of VHL, HIF-1α, CA9, and VEGF.', ['CA9', 'VHL'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Both up-regulated and down-regulated proteins were selected and identified as follows: heat shock protein 70 protein 2; Cyclin M3, alpha 1 antichymotrypsin precursor; carbonic anhydrase I, cancer antigen 1; SOBP protein; Rho GDP dissociation inhibitor alpha; glyoxalase I with benzyl glutathione inhibitor; lipid-free human apolipoprotein A-I and RAB4A member RAS oncogene family.', ['SOBP', 'RAB4A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We have studied the expression of toll-like receptors (TLR) TLR2, TLR4 and TLR9, and hypoxia markers HIF-1alpha and Carbonic anhydrase IX (CAIX) in normal and inflamed pancreatic ducts, in PanINs and in cancers.', ['Carbonic anhydrase IX', 'TLR2', 'TLR4', 'TLR9'])\n",
      "(' We found TLR2, TLR9, HIF-1alpha and CAIX to be prominently expressed in pancreatic intraepithelial neoplasia.', ['TLR2', 'TLR9'])\n",
      "(' Expression of TLR2 showed a linear increase from PanIN1 to PanIN3, while the highest TLR4 expression was detected in inflamed ducts, and TLR9 expression in PanIN1 lesions.', ['TLR2', 'TLR4', 'TLR9'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The present study aimed to determine the role of CA4 in GC tumorigenesis and its underlying molecular mechanism.', ['CA4', 'GC'])\n",
      "(' The levels of CA4 in GC cells and tissues were evaluated by reverse transcription-quantitative PCR and immunohistochemistry.', ['CA4', 'GC'])\n",
      "(' CA4 expression was suppressed in GC cells and tissues compared with adjacent healthy tissues and normal human gastric epithelial cells, respectively.', ['CA4', 'GC'])\n",
      "(' Analyses with a real-time cell analyzer and clonogenic assays were conducted to validate the impact of CA4 on GC cell lines (AGS and HGC-27) and normal human gastric epithelial cell line (GES-1) proliferation.', ['CA4', 'GC'])\n",
      "(' The effects of CA4 on the cell cycle in GC cells were determined by flow cytometry.', ['CA4', 'GC'])\n",
      "(' CA4 overexpression inhibited GC cell proliferation and reduced colony-forming ability, arrested the cell cycle in the G2/M phase, inhibited cyclin B1 and cyclin-dependent kinase 2 expression and induced p21 expression.', ['CA4', 'GC'])\n",
      "(' These results indicate that CA4 may serve an important role in GC tumorigenesis by inhibiting cellular proliferation via regulating the expression of cell cycle-associated proteins.', ['CA4', 'GC'])\n",
      "(' CA4 may serve as a diagnostic biomarker and a potential therapeutic target in GC.', ['CA4', 'GC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' CA9 suppression by inhibitors (S4 and U104) decreased viability and migration of MM cells, accompanied by overexpression of TFRC, IREB1/2 and FPN1(SLC40A1) and by downregulation of FTH/FTL.', ['CA9', 'TFRC'])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "(' Among these intersection genes, CA9, EBI3, SPOCK2, WDFY4, CD1D, and CCL22 were significantly correlated with overall survival in PDAC patients.', ['WDFY4', 'CD1D', 'CA9', 'SPOCK2', 'EBI3', 'CCL22'])\n",
      "(' Moreover, multivariate Cox analysis revealed that CA9, SPOCK2, and CD1D were the most significant prognostic genes, and were closely correlated with immune infiltration in TCGA cohort.', ['CA9', 'CD1D', 'SPOCK2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(\" Moreover, expression of four methylated genes (PLCXD3 (Phosphatidylinositol-Specific Phospholipase C, X Domain Containing 3), BAIAP2L2 (BAR/IMD Domain Containing Adaptor Protein 2 Like 2), NPR3 (Natriuretic Peptide Receptor 3), SNX10 (Sorting Nexin 10)) can influence patients' prognosis.\", ['SNX10', 'NPR3', 'BAIAP2L2'])\n",
      "(' Protein products of nine genes were identified as potential druggable targets, of which the tumorigenic potential of XDH (Xanthine Dehydrogenase), ATIC (5-Aminoimidazole-4-Carboxamide Ribonucleotide Formyltransferase/IMP Cyclohydrolase), CA9 (Carbonic Anhydrase 9), SLC7A11 (Solute Carrier Family 7 Member 11), and GAPDH (Glyceraldehyde-3-Phosphate Dehydrogenase) have been demonstrated in previous studies.', ['ATIC', 'SLC7A11', 'GAPDH', 'CA9', 'XDH'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' The present study aimed to evaluate the effects of E7070 on CA9 and CA12 enzymes in GBM cells as well as in the tumor cell growth, migration, invasion, and resistance to radiotherapy and chemotherapy.', ['CA9', 'CA12'])\n",
      "(' We showed that CA9 and CA12 represent potentially valuable therapeutic targets that should be further investigated as useful diagnostic and prognostic biomarkers for GBM tailored therapy.', ['CA9', 'CA12'])\n",
      "---------------------\n",
      "(' The expression of CA1 mRNA strongly correlated with that of sucrase-isomaltase, and inversely with that of MUC5AC in isolated IM glands.', ['MUC5AC', 'CA1'])\n",
      "(' In conclusion, CA1 could be colocalized with small intestinal proteins such as sucrase, but only rarely with the gastric mucin, MUC5AC.', ['MUC5AC', 'CA1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Finally, it inhibited genes and/or proteins expression of TGF-β1, 2 and 3, SMAD2, SMAD4, Wnt4, β-Catenin, LRP6, AXIN2 and Cyclin D1 in UL stem cells, all are key drivers of the TGF-β3/SMAD2 and Wnt4/β-Catenin pathways.', ['LRP6', 'SMAD4', 'AXIN2', 'SMAD2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We have previously demonstrated that therapeutically relevant concentrations/doses of oxaliplatin lead to a reduction in intracellular pH in mouse dorsal root ganglion (DRG) neurons in vitro and in vivo and that this alteration sensitizes TRPA1 and TRPV1 channels, which most likely mediate the allodynia associated with treatment.', ['TRPA1', 'TRPV1'])\n",
      "(' Furthermore, we show that FDA-approved drugs that inhibit carbonic anhydrase (an enzyme that is linked to haemoglobin in intracellular pH homeostasis), ie, topiramate and acetazolamide, revert (1) oxaliplatin-induced cytosolic acidification and TRPA1 and TRPV1 modulation in DRG neurons in culture, (2) oxaliplatin-induced cytosolic acidification of DRG of treated animals, and (3) oxaliplatin-induced acute cold allodynia in mice while not affecting OHP-induced cytotoxicity on cancer cells.', ['TRPA1', 'TRPV1'])\n",
      "---------------------\n",
      "(' The two most commonly involved genes are FGFR1 and CHD7.', ['FGFR1', 'CHD7'])\n",
      "(' In eugonadal disorders, activating FSHR mutations have been identified for spontaneous ovarian hyperstimulation syndrome; and WNT4 mutations have been described in mullerian aplasia.', ['FSHR', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Using tissue microarray, IHC staining for CA9 and CA12 was performed.', ['CA9', 'CA12'])\n",
      "(' Lymphatic invasion, the Ki-67 labelling index of primary tumours, and CA9 staining of stromal cells, were independent risk factors for SLN metastasis in the multivariate analysis.', ['CA9', 'SLN'])\n",
      "(' In 80 patients without lymphatic invasion of primary tumours or CA9 staining of stromal cells, only four patients (5%) had positive SLN.', ['CA9', 'SLN'])\n",
      "('CONCLUSION: CA9 staining of stromal cells is an independent risk factor for SLN metastasis as well as lymphatic invasion and a low Ki-67 labelling index of primary tumours in patients with early breast cancer.', ['CA9', 'SLN'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The expression level of NF-κB p65, NF-κB p50, HIF-1, HIF-2, growth factor VEGF, VEGFR2, CAIX, as well as AR, ERα, ERβ, PR, Brn-3α, TRIM16 were quantified by RT-PCR.', ['AR', 'TRIM16'])\n",
      "(' Its mRNA level depended on the expression of AR, Brn-3α, and TRIM16, components of the AKT/mTOR signaling pathway, transcription, and growth factors.', ['AR', 'TRIM16'])\n",
      "(' We revealed the growth in PR expression was accompanied by the suppression of the signaling cascade activity, AR, Brn-3α mRNA level, and the enhanced PTEN expression in PCa tissues.', ['PTEN', 'AR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Owing to the emergence of additional markers for RCCs, we have evaluated the immunohistochemical staining patterns of a series of 7 papillary cystadenomas using CK7, RCC, PAX8, carbonic anhydrase IX, and AMACR.', ['PAX8', 'AMACR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('2, we identify four more variants with statistically significant evidence of involvement in both endometriosis and WHRadjBMI (in/near KIFAP3, CAB39L, WNT4, GRB14); two of these, KIFAP3 and CAB39L, are novel associations for both traits.', ['CAB39L', 'KIFAP3', 'WNT4'])\n",
      "(' KIFAP3, WNT4 and 7p15.', ['KIFAP3', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Gene expression of Wnt pathway activator (WNT4), inhibitors (SFRP1, DKK3, AXIN1, and APC), transcriptional activator (TCF7), target genes (MYC, WISP2, and, CDH1), and Wnt modulator (TP53) was evaluated by qPCR.', ['AXIN1', 'DKK3'])\n",
      "(' The transcription levels of all genes studied, except APC, were higher in aCPs as compared to controls and TCF7 mRNA levels correlated with CTNNB1 mutation.', ['CTNNB1', 'APC', 'TCF7'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Here we studied the expression of ARG2 in pancreatic ductal carcinoma (PDC) tissue clinicopathologically by examining over 200 cases of PDC.', ['PDC', 'ARG2'])\n",
      "(' In contrast to prostate cancer, ARG2 expression was rarely demonstrated in PDC cells by immunohistochemistry, and instead ARG2 was characteristically expressed in α-smooth muscle actin-positive cancer-associated fibroblasts (CAFs), especially those located within and around necrotic areas in PDC.', ['PDC', 'ARG2'])\n",
      "(' In addition to the characteristic distribution of ARG2-expressing CAFs, such CAFs co-expressed carbonic anhydrase IX, SLC2A1, or HIF-1α, markers of hypoxia, in PDC tissue.', ['SLC2A1', 'PDC'])\n",
      "(' Furthermore, in vitro experiments revealed that cultured fibroblasts extracted from PDC tissue expressed the ARG2 transcript after exposure to hypoxia, which had arginase activity.', ['PDC', 'ARG2'])\n",
      "(' These results indicate that cancer cell-mediated immune suppression through ARG2 expression is not a general event and that the presence of ARG2-expressing CAFs is an indicator of poor prognosis, as well as hypoxia, in PDC tissue.', ['PDC', 'ARG2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' By searching the TCGA database, we showed that CA9, CXCL9, and GIMAP7 from the 8-mRNA signature were associated with the infiltration levels of immunocytes by regulating FOXO1 expression in PAAD.', ['CA9', 'FOXO1', 'CXCL9', 'GIMAP7'])\n",
      "(' Most importantly, we identified CA9, CXCL9, and GIMAP7 from the above eight genes as regulators of immunocyte infiltration by adjusting the expression of FOXO1 in PAAD.', ['CA9', 'FOXO1', 'CXCL9', 'GIMAP7'])\n",
      "(' Thus, CA9, CXCL9, and GIMAP7 might be the ideal targets of immune therapy of PAAD.', ['CA9', 'CXCL9', 'GIMAP7'])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Furthermore, inhibition of EGFR signaling intermediaries WNK1 and AKT1S1, members identified in the kinase array and previously unreported to play a role in the endometrium, also attenuate decidualization.', ['EGFR', 'AKT1S1', 'WNK1'])\n",
      "---------------------\n",
      "(' One tumor each had somatic FH mutation and SMARCB1 mutation.', ['FH', 'SMARCB1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' DNA was isolated from peripheral blood and used to genotype functional polymorphisms at the corresponding genes (HIF1A +1772 C\\u2009>\\u2009T, rs11549465; CA9\\u2009+\\u2009201 A\\u2009>\\u2009G; rs2071676; LOX +473\\xa0G\\u2009>\\u2009A, rs1800449; KDR - 604\\xa0T\\u2009>\\u2009C, rs2071559).', ['KDR', 'LOX'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We validated 27 genes using qRT-PCR and found their expression patterns were mostly similar to those of RNA-seq, including 12 novel ICM-dominant (HNF4A, CCL24, FGFR4, IFITM3, PTCHD2, GJB5, FN1, KLK7, PRDM14, GRP, FGF19 and GCM1) and two novel TE-dominant (SLC10A1 and WNT4) genes.', ['FGF19', 'KLK7', 'FN1', 'PRDM14', 'FGFR4', 'IFITM3', 'GRP', 'GJB5', 'CCL24'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' MVD and CA-9 expression were assessed immunohistologically by using the CD34 antibody and the MN/CA9 M75 antibody, respectively.', ['MVD', 'CD34'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In 46,XY gonadal dysgenesis, lack of testis development may be triggered by sex determining region Y, NR5A1, DHH or testis-determining gene loss-of-function mutations, DAX1 or WNT4 duplication or MAP3K1 gain-of-function mutations.', ['DHH', 'WNT4', 'MAP3K1', 'NR5A1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In the current study, we investigated the expression profiles of WNT/β-catenin signalling pathway genes (AXIN2, FZD4, TCF4, WNT3, WNT4, WNT5A, WNT7A, WNT1, APC, GSK3B and β-catenin) in a total of 13 oocyte samples.', ['FZD4', 'WNT4', 'GSK3B', 'WNT7A', 'TCF4', 'WNT3', 'WNT5A', 'APC', 'WNT1'])\n",
      "(' The results of this study displayed the absence of expression of AXIN2, FZD4, TCF4, WNT5A, WNT3, WNT4 and WNT7A genes in ovaries from women with PCOS and from healthy women.', ['FZD4', 'WNT4', 'AXIN2', 'WNT7A', 'TCF4', 'WNT3', 'WNT5A'])\n",
      "(' While APC and β-catenin expression levels were similar in the oocytes of both patients and controls, conversely, WNT1 and GSK3β genes both showed elevated expression in the oocytes of patients with PCOS, therefore suggesting an association between aberrant expression of WNT1 and GSK3β and the pathogenesis of PCOS.', ['WNT1', 'APC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Fusion-positive tumors had intense MAML3 nuclear staining and increased β-catenin by IHC and showed increased WNT4 expression.', ['WNT4', 'MAML3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('METHODS: A series of 98 tissue samples of primary non-small-cell lung carcinomas (NSCLC) from patients treated with surgery were analysed for the expression of CA9 and programmed-death ligand PD-L1 by cancer cells, and of FOXP3 by tumour-infiltrating lymphocytes (TILs).', ['CA9', 'FOXP3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' EGF did not stimulate tyrosine phosphorylation of CAIX, although EGFR and down-stream signaling pathways were activated by EGF.', ['EGFR', 'EGF'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(', PROM1, CD44, and EPCAM), it is found that CD44 positive population may render drug resistance in HCC272.', ['CD44', 'PROM1'])\n",
      "(' CellphoneDB analysis further implies that metabolism advantage organoids with enrichment of hypoxia signal upregulate NEAT1 expression in CD44 subgroup and mediate drug resistance that relies on Jak-STAT pathway.', ['CD44', 'NEAT1'])\n",
      "(' Moreover, metabolism advantage clusters shared in several organoids have similar characteristic genes (GAPDH, NDRG1 (N-Myc downstream regulated 1), ALDOA, and CA9).', ['NDRG1', 'ALDOA'])\n",
      "(' The combination of GAPDH and NDRG1 is an independent risk factor and predictor for patient survival.', ['NDRG1', 'GAPDH'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Nickel exposure increased the level of H3K4 trimethylation in both the promoters and coding regions of several genes including CA9 and NDRG1 that were increased in expression in A549 cells.', ['CA9', 'NDRG1'])\n",
      "(' We have also compared the extent of the H3K4 trimethylation in the absence and presence of formaldehyde crosslinking and observed that crosslinking of chromatin was required to observe H3K4 trimethylation in the coding regions immediately downstream of TSSs of some nickel-induced genes including ADM and IGFBP3.', ['ADM', 'IGFBP3'])\n",
      "---------------------\n",
      "(' We found that both CYP4B1 and RAB25 are downregulated with promoter hypermethylation and CA9 is upregulated with promoter hypomethylation, and we validated their mRNA differential expressions in 19 ccRCCs and 10 GEO datasets.', ['CA9', 'CYP4B1', 'RAB25'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(', PLK1, CXCR4, and AXL.', ['CXCR4', 'AXL', 'PLK1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Activation of MT1G also inhibited CA9 transcription through the suppression of HIF1A as mediated by KLF4.', ['CA9', 'HIF1A', 'KLF4', 'MT1G'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The spectrum of mutations in the NZBR lines was generally similar to that found in commonly used breast cancer cell lines but TP53 mutations were absent and mutations in EVI2B, LRP1B, and PMS2, which have not been reported in other breast cancer lines, were detected.', ['TP53', 'LRP1B', 'PMS2', 'EVI2B'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Our data support a model whereby FGF8 and WNT4 function in concert to induce the expression of Lim1 for MM survival and tubulogenesis.', ['FGF8', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We measured levels and stability of the hypoxia-inducible factor 1 subunit alpha (HIF1A), endothelial PAS domain 1 protein (EPAS1, also called HIF2A), CA9, solute carrier family 16 member 4 (SLC16A4, also called MCT4), and SLC2A1 (also called GLUT1) by immunoblot analyses.', ['CA9', 'SLC2A1'])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('RESULTS: Under hypoxic conditions, PDAC cells had increased levels of HIF1A and HIF2A, upregulated expression of CA9, and activated glycolysis.', ['CA9', 'HIF1A'])\n",
      "(' Knockdown of KRAS in PDAC cells, or incubation with trametinib, reduced the posttranscriptional stabilization of HIF1A and HIF2A, upregulation of CA9, pHi, and glycolysis in response to hypoxia.', ['CA9', 'HIF1A', 'KRAS'])\n",
      "('CONCLUSIONS: In response to hypoxia, PDAC cells that express activated KRAS increase expression of CA9, via stabilization of HIF1A and HIF2A, to regulate pH and glycolysis.', ['CA9', 'HIF1A', 'KRAS'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('METHODS: Nontumoral pancreas of 18 VHL patients operated on for PET, was examined for microadenomatosis (<or=5 mm) and compared with pancreatic specimen obtained from non-VHL patients or MEN1 patients.', ['VHL', 'MEN1'])\n",
      "(' The immunohistochemical expression of chromogranin, insulin, glucagon, HIF-1alpha, HIF-2alpha, VEGF, CA9, cyclin D1, and CD34 was assessed.', ['CA9', 'CD34'])\n",
      "(' Expression of HIF-1alpha and CA9 was not found in islets of controls and in MEN1 microadenomas.', ['CA9', 'MEN1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Downregulated PAX8 and upregulated EZH2 expression was found in GIST tissues.', ['PAX8', 'EZH2'])\n",
      "(' Overexpression of PAX8 or suppression of PAX8 methylation using DNA methyltransferase inhibitor 5-Aza-dC inhibited the proliferation, migration, and invasion of GIST cells while promoting their apoptosis (diminished PCNA, Ki67 and Bcl-2, elevated Bax, and cleaved caspase-3).', ['PAX8', 'PCNA'])\n",
      "(' EZH2 promoted PAX8 methylation to inhibit its expression.', ['PAX8', 'EZH2'])\n",
      "(' Collectively, EZH2 inhibits PAX8 expression by promoting its methylation, which thus downregulates Wnt4 expression, thereby promoting the development of GIST.', ['PAX8', 'EZH2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' NR2F1 expression was positively regulated by p38α signaling and repressed by HER2 and WNT4 pathways.', ['WNT4', 'NR2F1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('EXPERIMENTAL DESIGN: Expression of CA9 and CD34, an endothelial cell marker, was examined in serial paraffin-embedded sections by immunohistochemistry in 25 grade 1, 17 grade 2, and 20 grade 3 meningiomas.', ['CA9', 'CD34'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Besides, our results firstly identify that CREB can recruit SIRT1 (class III HDACS) by adaptor protein p300, then repress the expression of CA9.', ['CA9', 'SIRT1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Univariate Cox regression analysis revealed that six RNAs (2 DELs: FAM138B, SSTR5‑AS1; 2 DEMs: hsa‑miR‑149, hsa‑miR‑7; 2\\xa0DEGs: CCNE2, RRM2) were significantly associated with OS; while seven RNAs (1 DEL: LINC00284; 3 DEMs: hsa‑miR‑7, hsa‑miR‑15b, hsa‑miR‑30c‑2; and 3 DEGs: RRM2, CCNE2, CA2) were significantly associated with RFS.', ['FAM138B', 'CCNE2', 'RRM2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' EGF, TGF-alpha and secretin, but not CCK or GRP, bound to the TAKA-1 cells.', ['CCK', 'EGF', 'GRP'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('Poorer outcome in stromal HIF-2 alpha- and CA9-positive colorectal adenocarcinomas is associated with wild-type TP53 but not with BNIP3 promoter hypermethylation or apoptosis.', ['TP53', 'BNIP3'])\n",
      "(' Therefore, we correlated apoptosis, TP53 mutational status and BNIP3 promoter hypermethylation of CRC cells with HIF-2 alpha- and CA9-related poor outcome.', ['TP53', 'BNIP3'])\n",
      "(' In a series of 195 CRCs, TP53 mutations in exons 5-8 were analysed by direct sequencing, and promoter hypermethylation of BNIP3 was determined by methylation-specific PCR.', ['TP53', 'BNIP3'])\n",
      "(' Expressions of HIF-2 alpha, CA9, p53, BNIP3 and M30 were analysed immunohistochemically.', ['CA9', 'BNIP3'])\n",
      "(' Poorer survival of HIF-2 alpha and CA9 stromal-positive CRCs was associated with wild-type TP53 (P=0.', ['CA9', 'TP53'])\n",
      "(' Furthermore, apoptotic levels were independent of the TP53 status, but lower in unmethylated BNIP3 CRCs (P=0.', ['TP53', 'BNIP3'])\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: Expression of CA9, CSF1, DUSP6, STAT1, and MMD differed between NSCLC and normal lung.', ['MMD', 'DUSP6', 'CA9', 'CSF1', 'STAT1'])\n",
      "(' EGFR was more abundant in SCC compared with AC, whereas the reverse was true for DUSP6 and ERBB3.', ['EGFR', 'ERBB3', 'DUSP6'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Of note, few studies have reported the pathogenesis of CRC with the 3 identified hub genes, INHBA, CBX2 and BEST2.', ['BEST2', 'CBX2', 'INHBA'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Invasive cancer showed pooled expression rates of 35% for CAIX (95% confidence interval (CI): 26-46%), 51% for GLUT1 (CI: 40-61%), 46% for CXCR4 (CI: 33-59%), and 46% for IGF1R (CI: 35-70%).', ['CXCR4', 'IGF1R'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Moreover, the association was particularly high in a subgroup of basal-type BC patients with TP53 mutations thus revealing a putative cooperation of STC1 with mutated TP53 in generating highly aggressive BC subgroup.', ['TP53', 'STC1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Forty genes tested were expressed in NT2/D1 cells including the testis-determining genes SRY, SOX9, SF-1, DHH and FGF9.', ['DHH', 'SRY', 'SOX9', 'FGF9'])\n",
      "(' Genes not expressed included WT1, DAX1 and the ovary-specific genes FOXL2 and WNT4.', ['WT1', 'FOXL2', 'WNT4'])\n",
      "(' Our results suggest that male pathways initiated by SRY, SOX9 and SF-1 remain intact in these cells.', ['SRY', 'SOX9'])\n",
      "---------------------\n",
      "---------------------\n",
      "('9, p16, estrogen receptor, progesterone receptor, MUC6, PAX8, PAX2, p53, hepatocyte nuclear factor 1 beta, carbonic anhydrase IX, human epidermal receptor 2 (HER2), and mismatch repair (MMR) proteins.', ['PAX2', 'MUC6', 'PAX8'])\n",
      "(' There was positive staining with CK7 (47/47-45 diffuse, 2 focal), MUC6 (17/21-6 diffuse, 11 focal), carcinoembryonic antigen (25/31-12 diffuse, 13 focal), carbonic anhydrase IX (20/24-8 diffuse, 12 focal), PAX8 (32/47-20 diffuse, 12 focal), CA125 (36/45-5 diffuse, 31 focal), CA19.', ['MUC6', 'PAX8'])\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: Positive immunostaining was found for CK15, CK14, CDH3, NGFR p75, PEDF, Sox-2, Sox-10 and Wnt 4 in the basal dermoid epithelium, limbus and hair follicles.', ['NGFR', 'CDH3'])\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Regarding signaling patterns the majority of factors were expressed at similar levels in the subtypes, and only ANGPT2, MMP2, TIMP1, VEGFA and MMP9/TIMP2 were higher expressed in GBMs of the classical-like subtype.', ['TIMP1', 'MMP2', 'VEGFA', 'ANGPT2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: In clinical GBM samples, overexpression of S100A4 and NMIIA was observed in both non-pseudopalisading (Ps) and Ps (-associated) perinecrotic lesions, consistent with stabilization of HIF-1α and CA9.', ['CA9', 'S100A4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Of interest, exogenous FIH can rescue the ISWI-mediated upregulation of CA9 but not BNIP3, suggesting that FIH-independent mechanisms are also involved.', ['CA9', 'BNIP3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Mechanistically, LIF activates STAT, which binds to a Stat consensus sequence in the Six2 proximal promoter and sustains SIX2 levels.', ['LIF', 'SIX2'])\n",
      "(' Concomitantly, the combination of LIF/ROCKi upregulates Slug expression and activates YAP, which maintains SIX2, PAX2, and SALL1.', ['PAX2', 'SALL1', 'SIX2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' These results indicate that overexpression of HIF-1α and CA9 is associated with poor prognosis, and that HIF-1α overexpression is an independent unfavorable prognostic factor in STS.', ['CA9', 'STS'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The mRNA of Wnt4, Fzd3, sFRP4, APC and axin2 were downregulated in MM relative to mesothelial cells while LEF1 was overexpressed in MM.', ['APC', 'LEF1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('CONCLUSIONS: The following markers may have diagnostic utility in various diagnostic contexts: cytokeratins, vimentin, α-methylacyl coenzyme A racemase, carbonic anhydrase IX, PAX2, PAX8, RCC marker, CD10, E-cadherin, kidney-specific cadherin, parvalbumin, claudin-7, claudin-8, S100A1, CD82, CD117, TFE3, thrombomodulin, uroplakin III, p63, and S100P.', ['PAX8', 'S100A1', 'CD82', 'S100P', 'PAX2', 'TFE3'])\n",
      "(' PAX2 and PAX8 are sensitive and relatively specific markers for renal neoplasm, regardless of subtype.', ['PAX8', 'PAX2'])\n",
      "(' The potentially difficult differential diagnosis between chromophobe RCC and oncocytoma may be facilitated by S100A1 and CD82.', ['S100A1', 'CD82'])\n",
      "(' Together with high-molecular-weight cytokeratins, PAX2, and PAX8, they can help differentiate renal pelvic urothelial carcinoma from collecting duct RCC.', ['PAX8', 'PAX2'])\n",
      "(' Recognizing the variable sensitivity and specificity of these markers, it is important to include at least CD10, RCC marker, PAX2, and PAX8 in the diagnostic panel.', ['PAX8', 'PAX2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' SLC2A1 and VEGFA are promising novel prognostic factors in pediatric MPNST.', ['SLC2A1', 'VEGFA'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: Under normal pHe conditions, CA9, GLUT1 and LDHA gene expression was upregulated within 1-3h of anoxia, reaching near maximal values by 6h.', ['CA9', 'LDHA'])\n",
      "(' At 24 h the relative increase was 135, 12, 90 and 5 times for CA9, GLUT1, OPN and LDHA, respectively.', ['CA9', 'LDHA'])\n",
      "(' Gassing cells with differing oxygen concentrations for 24h resulted in a maximum level of expression for CA9 at 1% oxygen, whereas with GLUT1 and LDHA maximal expression occurred at 0.', ['CA9', 'LDHA'])\n",
      "(' Expression of CA9, GLUT1, OPN and LDHA was inhibited under hypoxic conditions when pHe was reduced to 6.', ['CA9', 'LDHA'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('MATERIALS AND METHODS: Genes of the WNT canonical pathway: activating ligands (WNT11, WNT4, WNT5A), binding inhibitors (DKK3, sFRP1), β-catenin (CTNNB1), β-catenin degradation complex (APC, AXIN1, GSK3β), inhibitor of β-catenin degradation complex (AKT1), sequester of β-catenin (CDH1), pathway effectors (TCF7, MAPK8, NFAT5), pathway mediators (DVL-1, DVL-2, DVL-3, PRICKLE, VANGL1), target genes (MYB, MYC, WISP2, SPRY1, TP53, CCND1); calcium dependent pathway (PLCB1, CAMK2A, PRKCA, CHP); and planar cell polarity pathway (PTK7, DAAM1, RHOA) were evaluated by QPCR, in 19 GH-, 18 ACTH-secreting, 21 non-secreting (NS) pituitary tumors, and 5 normal pituitaries.', ['TP53', 'WNT4', 'CAMK2A', 'SPRY1', 'DAAM1', 'MAPK8', 'MYC', 'AXIN1', 'PRKCA'])\n",
      "---------------------\n",
      "(' Elevated mRNA and protein levels of LRP5/6 and FZD10 paralleled expression of WNT2b and WNT4 in colonic crypts at days 6 and 12 post-infection with Citrobacter rodentium (CR) and tended to decline at days 20-34.', ['FZD10', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Exposure to excess norepinephrine significantly restricted the induced expression of decidualized markers Dtprp, BMP2, WNT4, and Hand2 in mice.', ['WNT4', 'BMP2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In conclusion, the coexpression of CA9 and HIF-1α in the absence of VHL gene abnormalities in CCPRCC suggests activation of the HIF pathway by mechanisms independent of VHL gene mutation.', ['CA9', 'VHL'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Five variants were previously reported to confer risk of various malignant or benign tumors (rs78378222 in TP53, rs10069690 in TERT, rs1800057 and rs1801516 in ATM, and rs7907606 at OBFC1) and four signals are located at established risk loci for hormone-related traits (endometriosis and breast cancer) at 1q36.', ['TP53', 'TERT', 'ATM'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In contrast with SDHA and SDHB immunohistochemistries that are routinely used to validate SDHx gene mutations, there is no such tool available for VHL mutations.', ['SDHA', 'VHL', 'SDHB'])\n",
      "(' The aim of this study was to evaluate whether CA9 immunostaining could be used as a tool to predict the presence or validate the pathogenicity of VHL gene mutations in paraganglioma.', ['CA9', 'VHL'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "(' SMG-1 is the newest and least studied member of the phosphoinositide 3-kinase-related kinase family, which consists of ATM, ATR, DNA-PKcs, mTOR, and SMG-1.', ['ATM', 'ATR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' IL2-XE114-TNFmut may thus be considered as a promising biopharmaceutical for the treatment of metastatic clear-cell renal cell carcinoma, since these tumors are known to be sensitive to IL2 and to TNF.', ['TNF', 'IL2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In a 32D cell-based system engineered to overexpress FZD2, FZD4, or FZD5, we found that WNT-3A (but not WNT-4, -5A, or -9B) activated the WNT-β-catenin pathway through FZD2/4/5 as measured by phosphorylation of LRP6 and β-catenin stabilization.', ['FZD2', 'FZD5', 'LRP6', 'FZD4'])\n",
      "(' Surprisingly, different WNT-FZD pairs showed differential effects on phosphorylation of DVL2 and DVL3, revealing a previously unappreciated DVL isoform selectivity by different WNT-FZD pairs in 32D cells.', ['DVL3', 'DVL2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Decreased expression of ID1, IGF1, GDF7, SMAD9, TGF-beta and WNT4, as well as GDF5, INHBA and ERBB4 overexpression, were confirmed in EC using RT-qPCR.', ['WNT4', 'ERBB4', 'ID1', 'GDF5', 'INHBA', 'IGF1', 'SMAD9', 'GDF7'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Furthermore, targets including WNT4, TGFB1, FOXO3, and a network of transcription factors were identified in the transitioning oocytes and somatic cells as potential upstream regulators that collectively may drive follicle activation.', ['TGFB1', 'WNT4', 'FOXO3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('The prognostic value of BAP1, PBRM1, pS6, PTEN, TGase2, PD-L1, CA9, PSMA, and Ki-67 tissue markers in localized renal cell carcinoma: A retrospective study of tissue microarrays using immunohistochemistry.', ['PBRM1', 'CA9', 'PTEN', 'BAP1'])\n",
      "('OBJECTIVE: To assess the prognostic roles of BAP1, PBRM1, pS6, PTEN, TGase2, PD-L1, CA9, PSMA, and Ki-67 tissue biomarkers in localized renal cell carcinoma (RCC).', ['PBRM1', 'CA9', 'PTEN', 'BAP1'])\n",
      "('CONCLUSION: pS6 and Ki-67 are significant prognostic factors of RCC; however, BAP1, PBRM1, TGase 2, PD-L1, CA9, PTEN loss, and PSMA markers did not show this association.', ['PBRM1', 'CA9', 'PTEN', 'BAP1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemical staining typically revealed both mononuclear and multinucleated cells to be positive for carbonic anhydrase IX, CD10, epithelial membrane antigen, vimentin, and cytokeratin AE1/AE3 and negative for β human chorionic gonadotropin, TFE3, cathepsin K, cytokeratin 7, cytokeratin 20, HMB45, CD68, smooth muscle actin, and S100.', ['CD68', 'TFE3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Pten(tm1Hwu/tm1Hwu);Ctnnb1(tm1Mmt/+);Amhr2(tm3(cre)Bhr/+) testicular tumors did not express CYP17, CYP19, germ cell nuclear antigen, estrogen receptor 1 or progesterone receptor, but expressed the early granulosa cell markers WNT4 and FOXL2, confirming the diagnosis of GCTT.', ['FOXL2', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Contrariwise, RFC1, KIF5C, ANXA2, N-RAP and RACK1 are overexpressed in GL261 cell the in the presence of the adipokines-enriched medium.', ['RACK1', 'KIF5C', 'RFC1', 'ANXA2'])\n",
      "(' We further identified the factors that are released by adipocyte cells, and revealed that several pro-inflammatory and angiogenic factors, such as IL-6, IL-11, LIF, PAI-1, TNF-α, endocan, HGF, VEGF IGF-I, were secreted to the medium into a high extent, whereas TIMP-1 and SerpinE1 were under expressed on CGl.', ['LIF', 'HGF'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('003) based on detection of BCL2, HER2, CD133, CAIX, and ERCC1.', ['BCL2', 'ERCC1'])\n",
      "('CONCLUSION: A proteomic panel of BCL2, HER2, CD133, CAIX, and ERCC1 independently predicted survival in locally advanced cervical cancer patients.', ['BCL2', 'ERCC1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('RESULTS: MiR-124 was over-expressed in SW620 cells, significantly reduced CCL20 and enhanced TNKS2 compared to that of the miR-NC group (p<0.', ['CCL20', 'TNKS2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Specifically, glioblastoma (GBM) CF had significantly lower GAL, HSPA5, and WNT4 levels than CSF from different grades of glioma.', ['GAL', 'WNT4', 'HSPA5'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('033, respectively) after ADT, whereas expression of EGFR, c-erbB-2, and CA9 was not influenced by ADT.', ['EGFR', 'CA9'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' High expression of the genes SPP1 and CSNK2A1 (encoding Osteopontin and CK2α respectively) correlated with poor prognosis while high expression of DEFB1 (encoding β-Defensin) correlated with better prognosis.', ['SPP1', 'DEFB1', 'CSNK2A1'])\n",
      "---------------------\n",
      "(' Stimulation of GOT1 cells by the CXCR4 ligand, CXCL12 (stromal cell-derived factor 1 (SDF-1)), activated the mitogen-activated protein kinase (MAPK) p42/44 signalling pathway and increased tumour cell migration.', ['CXCR4', 'CXCL12', 'GOT1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Likewise, application of either SFRP2 or SFRP5 on epithelial tumor cells completely abrogated PAR1-induced beta-catenin nuclear accumulation.', ['SFRP2', 'SFRP5'])\n",
      "---------------------\n",
      "('Prognostic role and correlation of CA9, CD31, CD68 and CD20 with the desmoplastic stroma in pancreatic ductal adenocarcinoma.', ['CA9', 'CD68'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We have found, in comparison with normal skin, that basal cell carcinomas have increased levels of mRNA for PTC1, GLI1, HIP, WNT2B, and WNT5a; decreased levels of mRNA for c-MYC, c-FOS, and WNT4; and unchanged levels of mRNA for PTC2, GLI2, WNT7B, and BMP2 and 4.', ['GLI2', 'WNT7B', 'WNT2B', 'GLI1', 'BMP2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Loss of PTEN increased cell migration, invasion, and upregulated WNT4, a key regulator of Müllerian duct development during embryogenesis.', ['WNT4', 'PTEN'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In addition to the general neuroendocrine cell biomarkers INSM1, synaptophysin, and chromogranins, these tumors are typically negative for keratins and instead have highly specific biomarkers, including the GATA3 transcription factor and enzymes involved in catecholamine biosynthesis: tyrosine hydroxylase that converts L-tyrosine to L-DOPA as the rate-limiting step in catecholamine biosynthesis, dopamine beta-hydroxylase that is present in cells expressing norepinephrine, and phenylethanolamine N-methyltransferase, which converts norepinephrine to epinephrine and therefore can be used to distinguish tumors that make epinephrine.', ['GATA3', 'INSM1'])\n",
      "(' In addition to these important tools that can be used to confirm the diagnosis of a paraganglioma, new tools are recommended to determine genetic predisposition syndromes; in addition to the identification of precursor lesions, molecular immunohistochemistry can serve to identify associations with SDHx, VHL, FH, MAX, and MEN1 mutations, as well as pseudohypoxia-related pathogenesis.', ['VHL', 'MAX', 'MEN1', 'FH'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' CDKN1A knockdown partially reversed the effects of WNT4 knockdown.', ['CDKN1A', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('Meta-analysis of whole-genome gene expression datasets assessing the effects of IDH1 and IDH2 mutations in isogenic disease models.', ['IDH2', 'IDH1'])\n",
      "---------------------\n",
      "('Tuberous sclerosis complex (TSC) is caused by mutations in either TSC1 or TSC2 genes and affects multiple organs, including kidney, lung, and brain.', ['TSC1', 'TSC2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Here, we assessed the actual impact of IDH1 and IDH2 mutations in patients harboring WHO grade II and III gliomas.', ['IDH2', 'IDH1'])\n",
      "(' We sequenced IDH1 at codon 132 and IDH2 at codon 172 in 33 patients with WHO grade II and III gliomas who benefited from a preoperative (18)F-FDG positron emission tomography (PET).', ['IDH2', 'IDH1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The adenocarcinoma component was strongly positive for cytokeratin 7 and pancytokeratin, weakly positive for synaptophysin and CD56, and negative for carbonic anhydrase IX, CD99, CDX2, chromogranin, cytokeratin 20 and smooth muscle actin.', ['CD99', 'CDX2'])\n",
      "(' The carcinoid tumor component was strongly positive for CD56, chromogranin and synaptophysin, weakly positive for pancytokeratin, and negative for carbonic anhydrase IX, CD99, CDX2, cytokeratin 7, cytokeratin 20 and smooth muscle actin.', ['CD99', 'CDX2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' CA12 shares a role with CA9 in acidification of micromilieu but it is less strictly regulated by hypoxia than CA9.', ['CA9', 'CA12'])\n",
      "(' In this study, we investigated expression of CA9 and CA12 mRNA in primary cervical cancer.', ['CA9', 'CA12'])\n",
      "('RESULTS: CA9 and CA12 mRNA expression was detected in 62.', ['CA9', 'CA12'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In parallel, there was increased expression of Wnt4 and β-Catenin, SOX9, and YAP1.', ['YAP1', 'SOX9'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In RCC patients with high leptin levels, the planar cell polarity (PCP)/JNK signaling pathway was shown to be activated, and genes in the axis, including CTHRC1, FZD2, FZD10, ROR2, WNT2, WNT4, WNT10B, WNT5A, WNT5B, and WNT7B, were upregulated.', ['WNT4', 'ROR2', 'WNT5B', 'CTHRC1', 'WNT7B', 'WNT2', 'WNT5A', 'WNT10B', 'FZD2', 'FZD10'])\n",
      "(' WNT5A and ROR2 are pivotal upstream regulators of PCP/JNK signaling, and their correlations with leptin expression levels were displayed by a Pearson correlation analysis.', ['ROR2', 'WNT5A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Thus, HIF-mediated increased expression of ARC in RCC can explain how loss of VHL can promote survival early in tumor formation.', ['VHL', 'ARC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' ADT induced the secretion of WNT4 which upon engagement of TCF7L1 in prostate cancer cells, enhanced IL-8 and CXCR2 expressions.', ['CXCR2', 'WNT4', 'TCF7L1'])\n",
      "(' TCF7L1 directly binds to the regulatory sequence region of IL-8 and CXCR2 through WNT4 activation, thus upregulating IL-8/CXCR2 signaling-driven NED and cell motility.', ['CXCR2', 'WNT4', 'TCF7L1'])\n",
      "(' Analysis of prostate tissue samples collected from small-cell neuroendocrine prostate cancer (SCPC) and castration-resistant prostate cancer (CRPC) tumors showed an increased intensity of nuclear TCF7L1 associated with CXCR2.', ['CXCR2', 'TCF7L1'])\n",
      "---------------------\n",
      "(' qRT-PCR analyses were conducted on 20 reflux- and inflammation-related genes, including interleukins 6 and 8, cytokeratins 8 and 14, mucin genes MUC1, MUC2, MUC3B, MUC4, MUC5B, MUC6 and MUC7 and carbonic anhydrase III.', ['MUC2', 'MUC5B', 'MUC7', 'MUC6', 'MUC4', 'MUC1'])\n",
      "('differences were noted in genes related to mucosal defence and inflammation, including CRNN, CD1d, TGFβ-1, MUC2, MUC5B and CDH1.', ['CDH1', 'MUC2', 'CRNN', 'MUC5B'])\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The expression levels of DEC2, CA9, and SLC2A1 were significantly higher in the 4 atypical hyperplasia tissues and 82 endometrial carcinomas compared with those in the 21 normal endometria.', ['CA9', 'SLC2A1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In a total of 80 cytogenetically characterized fibroids from 50 patients, we were not only able to confirm the frequent occurrence of MED12 mutations but also to stratify two mutually exclusive pathways of leiomyomagenesis with either rearrangements of HMGA2 reflected by clonal chromosome abnormalities affecting 12q14~15 or by mutations affecting exon 2 of MED12.', ['HMGA2', 'MED12'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' By immunohistochemistry, tumor cells were negative for SDHB (11/11) and rarely SDHA (1/11).', ['SDHA', 'SDHB'])\n",
      "(' Labeling was uniformly positive for PAX8 and kidney-specific cadherin and absent for KIT, RCC, and carbonic anhydrase IX.', ['PAX8', 'KIT'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Using chromatin immunoprecipitation we show that the eIF2alpha-dependent translationally regulated gene ATF4 binds directly to the CA9 promoter and is associated with loss of the transcriptional repressive histone 3 lysine 27 tri-methylation mark.', ['CA9', 'ATF4'])\n",
      "(' Loss or overexpression of ATF4 confirmed its role in CA9 induction during hypoxia.', ['CA9', 'ATF4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Using primary cultures of mouse OSE cells, we found WNT5a, but not WNT4, promotes EMT through a non-canonical Ca2+-dependent pathway, up-regulating the expression of Vimentin and CD44, enhancing cell migration, and inhibiting the CTNNB1 pathway and proliferation.', ['CD44', 'WNT4', 'CTNNB1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Lastly, MMP2 expression was increased by treatment with CA9 exosomes in HUVEC.', ['CA9', 'MMP2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' high levels of Ndufa4l2), lipid accumulation, and the expression of ccRCC markers ENO1 and CA9, and demonstrate that lipid accumulation and ccRCC development can potentially be attenuated by inhibiting Ndufa4l2.', ['CA9', 'ENO1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The pattern of association at these loci is consistent in BRCA1 and BRCA2 mutation carriers who are at high risk of EOC.', ['BRCA2', 'BRCA1'])\n",
      "(' After imputation to 1000 Genomes Project data, we assessed associations of 11 million genetic variants with EOC risk from 15,437 cases unselected for family history and 30,845 controls and from 15,252 BRCA1 mutation carriers and 8,211 BRCA2 mutation carriers (3,096 with ovarian cancer), and we combined the results in a meta-analysis.', ['BRCA2', 'BRCA1'])\n",
      "(' Incorporating these variants into risk assessment tools will improve clinical risk predictions for BRCA1 and BRCA2 mutation carriers.', ['BRCA2', 'BRCA1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Interaction networks of lncRNAs, miRNAs and six PCOS-related genes (IRS1, CYP11A1, BMP6, FSHR, WNT4 and CYP19A1) were constructed.', ['FSHR', 'CYP11A1', 'BMP6', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Markers such as CXCL16, ADAM10, B7-H1, Ki-67, survivin, P53, GLUT-1, calveolin-1 and endoglin are continuously being validated.', ['ADAM10', 'CXCL16'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('High expression of carbonic anhydrase IX is significantly associated with glandular lesions in gastroesophageal junction and with tumorigenesis markers BMI1, MCM4 and MCM7.', ['BMI1', 'MCM7', 'MCM4'])\n",
      "('METHODS: In this study, we investigated CA9 expression in esophageal cancers and in precancerous lesions and explored the association of CA9 expression with prognostic factors and with stem cell and tumorigenesis-related markers including BMI1, cyclin E, ki67, MCM4 and MCM7 expression.', ['CA9', 'BMI1', 'MCM7', 'MCM4'])\n",
      "(' Previously constructed tissue microarrays consisting of samples of 7 tissue types (columnar cell metaplasia, Barrett esophagus, low- and high-grade dysplasia, esophageal adenocarcinoma, squamous epithelium, and squamous cell carcinoma) were used for the immunostaining of CA9, BMI1, cyclin E, Ki67, MCM4 and MCM7.', ['CA9', 'BMI1', 'MCM7', 'MCM4'])\n",
      "(' High CA9 expression is significantly associated with BMI1, cyclin E, Ki67, MCM4 and MCM7 expression.', ['CA9', 'BMI1', 'MCM7', 'MCM4'])\n",
      "('CONCLUSIONS: High CA9 expression may be related to the acidic environment caused by gastroesophageal reflux disease in the gastroesophageal junction and associated with tumorigenesis through BMI1, MCM4 and MCM7.', ['CA9', 'BMI1', 'MCM7', 'MCM4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' The microarray analysis showed an 8-fold down-regulation in the expression of GLI1, WNT4 and WNT10A among itraconazole-treated CaSki cells.', ['GLI1', 'WNT4', 'WNT10A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' EGF receptor density (Receptor binding assay) was reduced by 50% within six hours and was reflected on abolishment of EGFR ligand-induced proliferation.', ['EGFR', 'EGF'])\n",
      "---------------------\n",
      "(' In addition, 3 PFD-CD did not significantly change the content of GSTP1 but produced an increase in its mRNA level (+20%), while it decreased the activities of both CAT (-66%) and GSTs (-26%).', ['GSTP1', 'CAT'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' SPC24, endothelial cell specific molecule 1, CDC20, CDCA3, cyclin (CCN) E1 and chromatin licensing and DNA replication factor 1 were significantly associated with cell division, mitotic cell cycle and positive regulation of cell proliferation.', ['CDCA3', 'SPC24', 'CDC20'])\n",
      "(' CCNB1, CCNE1, CCNB2 and stratifin were clearly associated with the p53 signaling pathway.', ['CCNB1', 'CCNB2', 'CCNE1'])\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(' We have used PeCa tissue arrays and multi-fluorophore labelled, quantitative, immunohistochemistry to interrogate the expression of WNT4, a Wnt ligand, and three targets of Wnt-ß-catenin transcription activation, namely, MMP7, cyclinD1 (CD1) and c-MYC in 141 penile tissue cores from 101 unique samples.', ['MMP7', 'WNT4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Both CA9 and CA12 were strongly induced by hypoxia in a range of tumor cell lines.', ['CA9', 'CA12'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Positivity of MCT4 and PSPH was related to shorter disease-free survival, and SHMT1 was related to shorter overall survival (P < .', ['SHMT1', 'PSPH'])\n",
      "(' Overexpression of MCT4, PSPH, and SHMT1 was associated with poorer prognosis.', ['SHMT1', 'PSPH'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('CONCLUSIONS: This study provides confirmatory data on the potential use of CA9 and TNFAIP6 as biomarkers of ccRCC.', ['CA9', 'TNFAIP6'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('MATERIALS AND METHODS: PD-L1/protein expression was evaluated in 295 resected NSCLCs and its correlation with HIF1A, VEGFA, GLUT1, CAIX expression and survival was determined based on immunohistochemical and RNA sequencing data obtained from The Cancer Genome Atlas (TCGA) database.', ['HIF1A', 'VEGFA'])\n",
      "('RESULTS: PD-L1 protein expression was significantly correlated with HIF1A, VEGFA, GLUT1, and CAIX expression only in adenocarcinoma when a 10% or a 50% cut-off was used.', ['HIF1A', 'VEGFA'])\n",
      "(' PD-L1 mRNA expression was also significantly correlated with HIF1A, VEGFA, GLUT1, and CAIX expression in adenocarcinoma.', ['HIF1A', 'VEGFA'])\n",
      "('CONCLUSIONS: PD-L1 protein and mRNA expression were correlated with HIF1A, VEGFA, GLUT1, and CAIX expression in adenocarcinoma alone.', ['HIF1A', 'VEGFA'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Expression of two major hypoxia-induced proteins, the hypoxia-inducible factor-1α (HIF1A) and the carbonic anhydrase IX (CA9), examined within a large cohort including 61 familial (3 BRCA1, 28 BRCA2, 30 BRCAX) and 225 sporadic male breast cancers showed that 31% of all male breast cancers expressed either HIF1A (25%) and/or CA9 (8%) in the combined cohort.', ['CA9', 'HIF1A', 'BRCA2', 'BRCA1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Here we show that treatment with DXR, PC, and CP induced upregulation of WNT4 expression, whereas WNT3 expression was downregulated by DXR and PC and upregulated by CP.', ['WNT3', 'PC', 'WNT4', 'CP'])\n",
      "(' Analysis of the WNT3 downstream signaling pathway showed that total β-catenin protein levels were reduced upon treatment with DXR and PC.', ['PC', 'WNT3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' SFRP1 protein is supposed to bind WNT1 molecules thereby inhibiting the activation of frizzled receptors and the WNT pathway.', ['WNT1', 'SFRP1'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' In PC3 and MDA-MB-231 cells, the mRNAs for COL1A1 and COL4A4 were up-regulated.', ['COL1A1', 'COL4A4'])\n",
      "(' The mRNAs for CTHRC1, LAMC2, and WNT7B were not changed when compared to the control.', ['WNT7B', 'LAMC2', 'CTHRC1'])\n",
      "---------------------\n",
      "(' Here we show that the onset of hypoxia-induced cancer stem cell features requires the beta-catenin-dependent post-transcriptional up-regulation of CA9 and SNAI2 gene expression.', ['CA9', 'SNAI2'])\n",
      "(' In response to hypoxia, beta-catenin moves from the plasma membrane to the cytoplasm where it binds and stabilizes SNAI2 and CA9 mRNAs, in cooperation with the mRNA stabilizing protein HuR.', ['CA9', 'SNAI2'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('METHODS: We performed Immunohistochemical staining of GLUT1, CAIX, BNIP3, p62, LC3, GLUD1, and GOT1.', ['GLUD1', 'GOT1', 'BNIP3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('PARTICIPANTS/MATERIALS SETTING METHODS: The changes in HIF1α, CA9, VEGFA and PR protein levels were measured by semi-quantitative analysis of immunostaining, and the expression levels of three genes (CA9, VEGFA and PR) were investigated by quantitative real-time PCR, in endometrial biopsies from 43 patients undergoing hysterectomy for EC (n\\u2009=\\u200922) or benign gynaecological indications (n\\u2009=\\u200921).', ['CA9', 'VEGFA'])\n",
      "---------------------\n",
      "(' Based on the indices of complementarity of CCAT1 and SNHG14 to the mRNA sequences, we selected 5 protein-coding genes MAPK1, c-MET, TGFB2, SNAIL1, and WNT4 associated with the epithelial-mesenchymal transition.', ['SNHG14', 'WNT4', 'TGFB2', 'MAPK1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Genes in the sub-pathway, such as RASSF1, PGK1, CA2 and CA12 were presumed to serve critical roles in laryngeal carcinoma.', ['RASSF1', 'CA2', 'PGK1', 'CA12'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We investigated the role of immunohistochemistry in the assessment of CA9, GLUT1, and PAX8 expression in VHL disease-related lesions.', ['CA9', 'VHL', 'PAX8'])\n",
      "(' Clinicopathologic information and archived pathology material from 5 patients with VHL disease were reviewed and evaluated for expression of\\xa0CA9, GLUT1, and PAX8.', ['VHL', 'PAX8'])\n",
      "(' CA9 and GLUT1 are consistently overexpressed in VHL disease-related lesions, reflecting upregulation of the HIF1a pathway.', ['CA9', 'VHL'])\n",
      "(' A combination of CA9 and GLUT1 immunostains is useful in screening lesions of patients with VHL spectrum manifestations, which may be targeted by the recently Food and Drug Administration-approved HIF-2a inhibitors.', ['CA9', 'VHL'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In the majority of cases in which immunostaining was observed in meningiomas, staining was focal (involving <5% of neoplastic cells) in comparison with CA9 and CA10 immunostaining in renal cell carcinomas in which more than 50% of tumor cells stained the majority of cases.', ['CA9', 'CA10'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('Aberrant methylation of MUC1 and MUC4 promoters are potential prognostic biomarkers for pancreatic ductal adenocarcinomas.', ['MUC4', 'MUC1'])\n",
      "(' MUC1 and MUC4 are high molecular weight transmembrane mucins.', ['MUC4', 'MUC1'])\n",
      "(' We evaluated the methylation status of MUC1 and MUC4 promoter regions in pancreatic tissue samples from 169 patients with various pancreatic lesions by the methylation specific electrophoresis (MSE) method.', ['MUC4', 'MUC1'])\n",
      "(' These results were compared with expression of MUC1 and MUC4, several DNA methylation/demethylation factors (e.', ['MUC4', 'MUC1'])\n",
      "(' We show that the DNA methylation status of the promoters of MUC1 and MUC4 in pancreatic tissue correlates with the expression of MUC1 and MUC4 mRNA.', ['MUC4', 'MUC1'])\n",
      "(' In addition, the expression of several DNA methylation/demethylation factors show a significant correlation with MUC1 and MUC4 methylation status.', ['MUC4', 'MUC1'])\n",
      "(' Furthermore, CAIX expression significantly correlates with the expression of MUC1 and MUC4.', ['MUC4', 'MUC1'])\n",
      "(' Interestingly, our results indicate that low methylation of MUC1 and/or MUC4 promoters correlates with decreased overall survival.', ['MUC4', 'MUC1'])\n",
      "(' This is the first report to show a relationship between MUC1 and/or MUC4 methylation status and prognosis.', ['MUC4', 'MUC1'])\n",
      "---------------------\n",
      "('TGF beta 2, LIF and FGF2 cooperate to induce nephrogenesis.', ['FGF2', 'LIF'])\n",
      "---------------------\n",
      "(' Numerous upregulated or downregulated genes were identified in diffuse-type GC, including CDH1, CDH2, VIM, WNT4 and WNT5.', ['WNT4', 'GC', 'CDH1', 'VIM', 'CDH2'])\n",
      "(' Among these genes, the mRNA ratio of CDH2 to CDH1 could distinguish the 15 diffuse-type GC samples from the 17 intestinal-type GC samples.', ['CDH1', 'CDH2', 'GC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('MATERIALS & METHODS: Expression and function of WNT4 and FZD6 were analyzed using qRT-PCR, Western blot, ELISA, in biopsies of non-neoplastic thymi (NT), thymoma and thymic carcinomas.', ['FZD6', 'WNT4'])\n",
      "(' Down-regulation of WNT4 by shRNA induced cell death in pTECs derived from B3 thymomas and led to decreased RAC1, but not JNK protein phosphorylation.', ['WNT4', 'RAC1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' TCF1 and TCF4 factors, and LRP5 and LRP6 co-receptors were ubiquitously expressed.', ['LRP6', 'TCF4', 'LRP5'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('L1CAM, CA9, KLK6, HPN, and ALDH1A1 as Potential Serum Markers in Primary and Metastatic Colorectal Cancer Screening.', ['L1CAM', 'CA9', 'KLK6', 'ALDH1A1', 'HPN'])\n",
      "(' HPN and KLK6 showed significant differences (p < 0.', ['KLK6', 'HPN'])\n",
      "(' In contrast, levels of HPN and ALDH1A1 were significantly elevated (p < 0.', ['HPN', 'ALDH1A1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Furthermore, the impact of nicotine treatment on phosphorylation of STAT3, HIF-1α and CA9 expression was assessed in HT29 cells.', ['CA9', 'STAT3'])\n",
      "---------------------\n",
      "('The expression of the hypoxia markers CA9 and CXCR4 is correlated with survival in patients with neuroendocrine tumours of the ileum.', ['CA9', 'CXCR4'])\n",
      "('BACKGROUND/AIM: The hypoxia-inducible factor pathway regulates the expression of a diverse group of molecules such as CA9 and CXCR4.', ['CA9', 'CXCR4'])\n",
      "('METHODS: The immunohistochemical expression of CA9 and CXCR4 was examined in 51 patients with a resected IET.', ['CA9', 'CXCR4'])\n",
      "(\" A 'hypoxic score' was calculated, integrating the expression of both CA9 and CXCR4 (hypoxic score 0: absence of expression of both molecules; hypoxic score 1: expression of CXCR4 and/or CA9).\", ['CA9', 'CXCR4'])\n",
      "('CONCLUSION: The hypoxia-inducible factors CA9 and CXCR4 were found associated to the malignant progression of neuroendocrine tumours of the ileum.', ['CA9', 'CXCR4'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('METHODS: Immunoperoxidase staining for FOXP3 and CXCL12 was performed on tissue microarrays from 491 breast cancers.', ['FOXP3', 'CXCL12'])\n",
      "('CONCLUSIONS: Our data show that in the setting of hypoxia and CXCR4 up-regulation in Treg, CXCL12 expression may have the negative consequence of enhancing Treg recruitment and suppressing the anti-tumour immune response.', ['CXCR4', 'CXCL12'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' CA9 and RGS5 expression levels of both groups were compared using immunohistochemistry, reverse transcription‑quantitative polymerase chain reaction and ELISA.', ['CA9', 'RGS5'])\n",
      "(' Compared with the control (NS) group, both CA9 and RGS5 expression in rh‑ES group were significantly decreased at days 4 and 6 (P<0.', ['CA9', 'RGS5'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Further analysis revealed that FZD2 is positively correlated with WNT3A, WNT5B, WNT7A and WNT2 and is negatively correlated with WNT4.', ['WNT4', 'WNT7A', 'WNT5B', 'WNT2', 'FZD2', 'WNT3A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Renal tumors with entity-defining genetic alterations may ultimately require molecular confirmation via cytogenetics or sequencing technologies, such as RCC with TFE3, TFEB, or ALK gene rearrangements, or TFEB amplified RCC.', ['TFEB', 'ALK', 'TFE3'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In Ishikawa cells, a human endometrial epithelial cell line, LEFTY2 downregulated the expression of calcium release-activated calcium channel protein 1, encoded by ORAI1, and inhibited store-operated Ca2+ entry (SOCE).', ['LEFTY2', 'ORAI1'])\n",
      "(' Furthermore, LEFTY2 and the Orai1 blockers 2-APB, MRS-1845, as well as YM-58483, inhibited, whereas the Ca2+ ionophore, ionomycin, strongly upregulated COX2, BMP2 and WNT4 expression in decidualizing HESCs.', ['LEFTY2', 'WNT4', 'BMP2'])\n",
      "(' •LEFTY2 and the Orai1 blockers 2-APB, MRS-1845, and YM-58483 inhibit COX2, BMP2, and WNT4 expression in endometrial cells.', ['WNT4', 'BMP2'])\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "(' Our results demonstrated that ALDH1, CD44, ARID1A, and CAIX in immunoreactive GA tumor cells exhibit different expression profiles compared with control cells and that these differences are associated with patient survival.', ['CD44', 'ARID1A'])\n",
      "(' In addition, the loss of ALDH1, ARID1A, and CD44 and the overexpression of CAIX are important for tumor invasion and metastasis; therefore, they may serve as useful prognostic indicators of long-term survival in patients with GA.', ['ARID1A', 'CD44'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Recently, we identified the tumour-associated carbonic anhydrases (CA), CA9 and CA12 as hypoxia-inducible in tumour cell lines.', ['CA9', 'CA12'])\n",
      "---------------------\n",
      "(' Linear regression analysis comparing FAZA retention with patterns of gene expression in individual tissue fragments revealed that CA9, GLUT1 and LOX mRNA levels were equally and strongly correlated to hypoxic extent in FaDudd.', ['CA9', 'LOX'])\n",
      "(' The same link between hypoxia and gene expression profile was observed for CA9 and GLUT1, but not LOX, in SCCVII tumors.', ['CA9', 'LOX'])\n",
      "(' Among tested genes, our study identified CA9, GLUT1 and possibly LOX as highly specific biomarkers of tumor hypoxia in vivo.', ['CA9', 'LOX'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' A combination of RENCA lysates and AbOmpA up-regulated the surface expression of co-stimulatory molecules, CD80 and CD86, and the antigen presenting molecules, major histocompatibility (MHC) class I and class II, in DCs.', ['CD80', 'CD86'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' This was in accord with very low expression of TSC1 (that is associated with HIF activation) and a high expression of TSC2.', ['TSC1', 'TSC2'])\n",
      "(' Moreover, angiomyolipoma specimens showed the overexpression of the ER stress markers XBP1, CHOP and ATF4 as well as of the mediators of calcium metabolism, namely the type 1 and 2, but not the type 3 IP3 receptors.', ['ATF4', 'XBP1'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' The increase in VHL expression was accompanied an increase in the level of mRNA for NF-κB p65 and kinases PDK1 and Akt.', ['VHL', 'PDK1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('The forkhead box (FOX), FOXO1 and FOXO3, transcription factors regulate multiple functions in mammalian cells.', ['FOXO1', 'FOXO3'])\n",
      "(' Immunostaining and Western blot analyses confirmed FOXO1 and phosphatase and tensin homolog (PTEN) depletion, maintenance of globin transcription factor (GATA) 4 and nuclear localization of FOXL2 and phosphorylated small mothers against decapentaplegic (SMAD) 2/3 in the tumor cells, recapitulating results we observed in human adult GCTs.', ['FOXO1', 'FOXL2'])\n",
      "(' Thus, analyses of the Foxo1/3/Pten mouse GCTs and human adult GCTs provide strong evidence that impaired functions of the FOXO1/3/PTEN pathways lead to dramatic changes in the molecular program within granulosa cells, chronic activin signaling in the presence of FOXL2 and GATA4, and tumor formation.', ['FOXL2', 'GATA4'])\n",
      "---------------------\n",
      "---------------------\n",
      "('The serum levels of a-PBP, IgG4, IgG, anti-nuclear antibodies (ANA), anti-lactoferrin (a-LF), a-CA-II, and rheumatoid factor (RF) were evaluated in patients with AIP (n\\u200a=\\u200a29), PC (n\\u200a=\\u200a17), pancreatic neuroendocrine neoplasm (P-NEN, n\\u200a=\\u200a12), and alcoholic chronic pancreatitis (ACP, n\\u200a=\\u200a41).', ['PC', 'AIP'])\n",
      "('There was no statistically significant difference in mean a-PBP values in AIP compared with PC.', ['PC', 'AIP'])\n",
      "('3\\u200aU, low values of a-PBP had a sensitivity and specificity of 45% and 71% for differentiating AIP from PC.', ['PC', 'AIP'])\n",
      "('4\\u200ag/L) for differentiating AIP from PC was 45% and 88%, but rose to 52% and 88% when using a cut-off of 1.', ['PC', 'AIP'])\n",
      "(' When using this cut-off, the sensitivity and specificity for differentiating type 1 AIP from PC was 68% and 88%.', ['PC', 'AIP'])\n",
      "(' None of the other markers were significantly changed in AIP versus PC.', ['PC', 'AIP'])\n",
      "('The only serological marker for which we found a statistically significant difference in mean values between AIP and PC was IgG4.', ['PC', 'AIP'])\n",
      "(' However, the value of IgG4 for the distinction of AIP from PC was limited, probably in part due to the relatively high number of type 2 AIP patients in our study.', ['PC', 'AIP'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We investigated whether these biomarkers (HIF-1α, HIF-2α, CAIX, VEGF, VEGFR1, VEGFR2, VEGFR3, PDGFB, PDGFRA, PDGFRB, CD31, CD44, bcl-xL, KIT, p21, CXCR4, PTEN, (CSF)-1R, RET, and FLT-3) can predictive the different effects between sunitinib and sorafenib treatments and are available to guide targeted drug selection.', ['PDGFB', 'PDGFRB', 'CXCR4', 'PTEN', 'CD44', 'PDGFRA', 'KIT', 'RET'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' From 9 patients without cirrhosis, 3 presented NASH F3 and 6 NASH F2.', ['F2', 'F3'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' The expression of Beclin-1, ARID1A, IDH1 and CA9 were determined by immunohistochemical staining.', ['ARID1A', 'CA9', 'IDH1'])\n",
      "('RESULTS: Beclin-1, ARID1A, CA9 and IDH1 were highly expressed in ICC tumor tissues.', ['ARID1A', 'CA9', 'IDH1'])\n",
      "(' Neither CA9 nor IDH1 expression was significantly associated with mortality or disease recurrence.', ['CA9', 'IDH1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' In other cancers, the activation of canonical wnt/beta-catenin signaling is associated with APC, CTNNB1 or AXIN1 mutations.', ['AXIN1', 'APC', 'CTNNB1'])\n",
      "(' HMEC expressed all Frizzled receptors except FZD9 and FZD10.', ['FZD10', 'FZD9'])\n",
      "(' They also expressed LRP5 and LRP6 co-receptors, as well as four Lef/TCF transcription factors.', ['LRP6', 'LRP5'])\n",
      "(' HMEC cells also expressed many Wnt ligands, including WNT1, WNT2B, WNT3, WNT5A, WNT5B, WNT7B, WNT9A, WNT10B and WNT16.', ['WNT5B', 'WNT3', 'WNT7B', 'WNT5A', 'WNT2B', 'WNT10B', 'WNT9A', 'WNT1', 'WNT16'])\n",
      "(' We noticed that most breast cancer cell lines overexpressed WNT3A, WNT4, WNT6, WNT8B, WNT9A and WNT10B.', ['WNT6', 'WNT4', 'WNT8B', 'WNT10B', 'WNT9A', 'WNT3A'])\n",
      "(' In contrast, the expression of WNT5A, WNT5B and WNT16 was usually down-regulated.', ['WNT5B', 'WNT5A', 'WNT16'])\n",
      "(' It is noteworthy that all six Wnt ligands that are overexpressed in malignant cell lines are known to signal through the canonical Wnt/beta-catenin signaling pathway, whereas down-regulated WNT5A and WNT5B ligands signal via the non-canonical pathway.', ['WNT5B', 'WNT5A'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "---------------------\n",
      "(' Antibodies were used to quantify endogenous POR, MTRR, NDOR1, and NOS2A expression in 23 human tumor cell lines; however, only POR protein was detectable and its expression correlated with anoxic PR-104A reduction (r(2) = 0.', ['MTRR', 'NDOR1', 'POR'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' FBXL14 mRNA is decreased in tumors with a high expression of two proteins up-regulated in hypoxia, carbonic anhydrase 9 and TWIST1.', ['FBXL14', 'TWIST1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "('MATERIALS AND METHODS: Expression of the hypoxia-induced genes OPN, CA9, EPO, VEGF and HIF-1alpha was analyzed in three GBM cell lines, GaMG, U373 and U251, under in vitro hypoxia (1, 6 or 24h at 5%, 1% or 0.', ['CA9', 'EPO'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemical stains for markers of HIF pathway activation (HIF-1α, CA9, and glucose transporter-1 (GLUT-1)) as well as other relevant markers (CK7, CD10, AMACR, and TFE3) were performed.', ['CA9', 'AMACR'])\n",
      "(' No expression of TFE3, CD10, or AMACR was seen.', ['TFE3', 'AMACR'])\n",
      "(' The co-expression of CA9, HIF-1α, and GLUT-1 in the absence of VHL gene alterations in clear-cell papillary renal cell carcinoma suggests activation of the HIF pathway by non-VHL-dependent mechanisms.', ['CA9', 'VHL'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' We show that the interaction with two of these proteins, namely XPO1 exportin and TNPO1 importin, occurs via the C-terminal region of CA IX and increases with protein phosphorylation.', ['XPO1', 'TNPO1'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemically, the tumor were positive for BRM, PBRM1, ARID1A, CD10, CKpan, Vimentin, carbonic anhydrase IX (CA-IX), and P504S (AMACR) but negative for INI1, BRG1, HMB45, melan A, CK7, CD117, Ksp-cadherin, TFEB, TFE3, and Cathepsin K.', ['PBRM1', 'TFEB', 'TFE3', 'ARID1A'])\n",
      "---------------------\n",
      "---------------------\n",
      "(' The expression of angiogenesis markers VEGF-A, VEGFR, PDGFββ, PDGFR, CCND1 and CA9 was assessed by immunohistochemistry (IHC) and correlated with overall survival (OS) and progression-free survival (PFS).', ['CA9', 'CCND1'])\n",
      "(' CCND1 and CA9 were positive in 44% and 60% of cases.', ['CA9', 'CCND1'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' However, the data showed that TSA down-regulated influential transcripts that drive the BRAF-extracellular signal-regulated kinase (ERK)1/2 oncogenic pathway (namely PRKCD and MYC proto-oncogene which negatively affected the cell-cycle distribution.', ['PRKCD', 'MYC'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' FISH and reverse-transcription polymerase chain reaction assays showed no EWSR1 and NR4A3 loci rearrangements.', ['EWSR1', 'NR4A3'])\n",
      "(' Microarray-based analysis also revealed no changes in NR4A3 and EWSR1 gene transcription levels.', ['EWSR1', 'NR4A3'])\n",
      "---------------------\n",
      "(' In this study, the effect of 10 cytokines on the expression of the mCRP CD46, CD55 and CD59 and the renal tumour-associated antigen G250/MN/CAIX on four human renal tumour cell lines and proximal tubular epithelial cells was determined by flow cytometry.', ['CD55', 'CD59', 'CD46'])\n",
      "(' In addition, it was measured whether changes in the expression of the classical pathway regulatory proteins CD55 and CD59 had an effect on C3 deposition and lysis.', ['CD59', 'C3', 'CD55'])\n",
      "(' Interleukin-1beta (IL-1beta) consistently downregulated the expression of CD46 and CD59; IL-4 consistently downregulated the expression of CD46 and transforming growth factor-beta1, consistently downregulated the expression of both CD46 and CD55.', ['CD55', 'CD46'])\n",
      "(' Changes in the expression of CD55 and CD59 were associated with changes in the amount of C3 deposited and the extent of complement-mediated lysis, respectively.', ['CD59', 'C3', 'CD55'])\n",
      "(' This suggests that clinical immunotherapy, consisting of treatment with cytokines and MoAb, may induce either up- or downregulation of CD55 or CD59 and thus affect the effectiveness of immunotherapy with MoAb.', ['CD59', 'CD55'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "(' Immunohistochemistry revealed positivity for Pan-CK, CK7, CK8/18, MUC1, MUC6, carbonic anhydrase IX, CD10, EMA, β-catenin and e-cadherin.', ['MUC6', 'MUC1'])\n",
      "(' Sanger sequencing did not detect mutations for β-catenin, BRAF, KRAS, PIK3CA and GNAS.', ['PIK3CA', 'KRAS', 'BRAF', 'GNAS'])\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n",
      "---------------------\n"
     ]
    }
   ],
   "source": [
    "\n",
    "result = []\n",
    "for item in data_train:\n",
    "#     print(item)\n",
    "    print('---------------------')\n",
    "    item_result = []\n",
    "    for line in item:\n",
    "        line_match = re.findall(gene_regex, line)\n",
    "        line_res = []\n",
    "        if len(line_match) >= 2:\n",
    "            for match in line_match:\n",
    "                match_str = match[0].strip()\n",
    "                match_str = match_str[:-1] if match_str[-1] in ['.', ','] else match_str\n",
    "                line_res.append(match_str)\n",
    "            line_res = list(set(line_res))\n",
    "            if len(line_res) >= 2:\n",
    "                item_result.append((line, line_res))\n",
    "                print((line, line_res))\n",
    "    result.append((item, item_result))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "8a95fa3e",
   "metadata": {},
   "outputs": [],
   "source": [
    "s='FSTX1\tSFTX1\tYSX1\tFXX1\t头晕\t黄疸\tPQTTX1\tBJQX1\t坐立不安\tHBX1\tYXFTX1\tZYXTX1\tZFTX1\tYSFTX1\tFSDFX1\tCXFTX1\tYHJZX1\tZFXTX1\t突然发作\t绞痛\tJLFTX1\t恶心呕吐\t腹胀\tTZPBX1\t停经\tYDLXX1\tPNGBX1\tFTSCFX1\tYCFTX1\t性别\t年龄'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "eb0831db",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31\n"
     ]
    }
   ],
   "source": [
    "print(len(s.split('\t')))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "af6ab212",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c25ef73a",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
